{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import subprocess\n",
    "from Bio import SeqIO\n",
    "import pickle\n",
    "import os\n",
    "from pybedtools import BedTool\n",
    "from multiprocessing import Pool\n",
    "from itertools import chain\n",
    "import h5py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "rnas = pd.read_csv(\"../all_marks/H3K27ac/our_fantom_genes_association_pvalues.tsv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "rnas_with_prefixes = []\n",
    "for i, row in rnas.iterrows():\n",
    "    if(row['pvalue'] < 0.05):\n",
    "        rnas_with_prefixes.append((row['lncRNAId'], '', ''))\n",
    "    if(row['mm_pvalue'] < 0.05):\n",
    "        rnas_with_prefixes.append((row['lncRNAId'], '', '_m'))\n",
    "        rnas_with_prefixes.append((row['lncRNAId'], '_m', 'm'))\n",
    "    if(row['mp_pvalue'] < 0.05):\n",
    "        rnas_with_prefixes.append((row['lncRNAId'], '', '_p'))\n",
    "        rnas_with_prefixes.append((row['lncRNAId'], '_m', 'p'))\n",
    "    if(row['pm_pvalue'] < 0.05):\n",
    "        rnas_with_prefixes.append((row['lncRNAId'], '', '_m'))\n",
    "        rnas_with_prefixes.append((row['lncRNAId'], '_p', 'm'))\n",
    "    if(row['pp_pvalue'] < 0.05):\n",
    "        rnas_with_prefixes.append((row['lncRNAId'], '', '_p'))\n",
    "        rnas_with_prefixes.append((row['lncRNAId'], '_p', 'p'))\n",
    "rnas_with_prefixes = set(rnas_with_prefixes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "92"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(rnas_with_prefixes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def generateTranscriptFiles():\n",
    "    fasta_sequences = SeqIO.parse(open(\"/home/ymedvedeva/FANTOM6/data/private/F6_CAT.DMFB.target.all_transcript.fa\"), 'fasta')\n",
    "    for fasta in fasta_sequences:\n",
    "        print(fasta.id.split('|')[1])\n",
    "        with open(\"../fantom6/transcripts/\" + fasta.id.split('|')[1] + \".fa\", \"w\") as output_handle:\n",
    "            SeqIO.write(fasta, output_handle, \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#generateFilesForTests(['ENSG00000231312', 'ENSG00000214548', 'ENSG00000246273', 'ENSG00000260032'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "#generate proms and background files for promoter and region tests\n",
    "def generateFilesForTests(target, rnas):\n",
    "    fantom = pd.read_csv(\"../fantom6/oligo_DE_Summary_gene_filtered.tsv\", sep=\"\\t\")\n",
    "    \n",
    "    #В качестве бэкграунда берем все деги(а не просто все фантомовские гены)\n",
    "    fantom_all_genes = set(fantom['geneID'].unique())\n",
    "    \n",
    "    #Уже проверено, что гены из общего списка\n",
    "    with open(\"../all_marks/\" + target + \"/our_fantom_genes_association.pickle\", 'rb') as f:\n",
    "        our_fantom_genes = pickle.load(f)\n",
    "        \n",
    "    if not os.path.exists(\"../all_marks/\" + target + \"/fantom_tdf/\"):\n",
    "        os.makedirs(\"../all_marks/\" + target + \"/fantom_tdf/\")\n",
    "        \n",
    "    fantom_ann = pd.read_csv(\"/home/ymedvedeva/FANTOM6/data/public/FANTOM5-CAT/F6_CAT.gene.info.tsv\", sep=\"\\t\")\n",
    "    fantom_ann = fantom_ann[['cntg', 'geneStart', 'geneEnd', 'geneID', 'geneName', 'strnd']]\n",
    "    \n",
    "    for rna in rnas:\n",
    "        #Все деги\n",
    "        genes = set(our_fantom_genes[rna]['fantom'])\n",
    "        #Все деги c положительным fc\n",
    "        genes_p = set(our_fantom_genes[rna]['fantom_plus'])\n",
    "        #Все деги c положительным fc\n",
    "        genes_m = set(our_fantom_genes[rna]['fantom_minus'])\n",
    "        #Все деги c положительным fc и положительной корреляцией\n",
    "        genes_pp = set(our_fantom_genes[rna]['fantom_plus']).intersection(set(our_fantom_genes[rna]['our_plus']))\n",
    "        #Все деги c положительным fc и отрицательной корреляцией\n",
    "        genes_pm = set(our_fantom_genes[rna]['fantom_minus']).intersection(set(our_fantom_genes[rna]['our_plus']))\n",
    "        #Все деги c отрицательным fc и положительной корреляцией\n",
    "        genes_mp = set(our_fantom_genes[rna]['fantom_plus']).intersection(set(our_fantom_genes[rna]['our_minus']))\n",
    "        #Все деги c отрицательным fc и отрицательной корреляцией\n",
    "        genes_mm = set(our_fantom_genes[rna]['fantom_minus']).intersection(set(our_fantom_genes[rna]['our_minus']))\n",
    "\n",
    "        background_genes = fantom_all_genes - genes\n",
    "        background_genes_p = fantom_all_genes - genes_p\n",
    "        background_genes_m = fantom_all_genes - genes_m\n",
    "        background_genes_pp = fantom_all_genes - genes_pp\n",
    "        background_genes_pm = fantom_all_genes - genes_pm\n",
    "        background_genes_mp = fantom_all_genes - genes_mp\n",
    "        background_genes_mm = fantom_all_genes - genes_mm\n",
    "\n",
    "        genes_bed = fantom_ann[fantom_ann['geneID'].isin(genes)]\n",
    "        background_genes_bed = fantom_ann[fantom_ann['geneID'].isin(background_genes)]\n",
    "\n",
    "        genes_bed_p = fantom_ann[fantom_ann['geneID'].isin(genes_p)]\n",
    "        background_genes_bed_p = fantom_ann[fantom_ann['geneID'].isin(background_genes_p)]\n",
    "        \n",
    "        genes_bed_m = fantom_ann[fantom_ann['geneID'].isin(genes_m)]\n",
    "        background_genes_bed_m = fantom_ann[fantom_ann['geneID'].isin(background_genes_m)]\n",
    "\n",
    "        genes_bed_pp = fantom_ann[fantom_ann['geneID'].isin(genes_pp)]\n",
    "        background_genes_bed_pp = fantom_ann[fantom_ann['geneID'].isin(background_genes_pp)]\n",
    "        \n",
    "        genes_bed_pm = fantom_ann[fantom_ann['geneID'].isin(genes_pm)]\n",
    "        background_genes_bed_pm = fantom_ann[fantom_ann['geneID'].isin(background_genes_pm)]\n",
    "        \n",
    "        genes_bed_mp = fantom_ann[fantom_ann['geneID'].isin(genes_mp)]\n",
    "        background_genes_bed_mp = fantom_ann[fantom_ann['geneID'].isin(background_genes_mp)]\n",
    "        \n",
    "        genes_bed_mm = fantom_ann[fantom_ann['geneID'].isin(genes_mm)]\n",
    "        background_genes_bed_mm = fantom_ann[fantom_ann['geneID'].isin(background_genes_mm)]\n",
    "\n",
    "        proms = getProms(genes_bed)\n",
    "        background_proms = getProms(background_genes_bed)\n",
    "\n",
    "        proms.to_csv(\"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_genes_proms.bed\", sep=\"\\t\", header=None, index=None)\n",
    "        background_proms.to_csv(\"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_background_genes_proms.bed\", sep=\"\\t\", header=None, index=None)\n",
    "\n",
    "        proms_p = getProms(genes_bed_p)\n",
    "        background_proms_p = getProms(background_genes_bed_p)\n",
    "\n",
    "        proms_p.to_csv(\"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_genes_p_proms.bed\", sep=\"\\t\", header=None, index=None)\n",
    "        background_proms_p.to_csv(\"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_background_genes_p_proms.bed\", sep=\"\\t\", header=None, index=None)\n",
    "        \n",
    "        proms_m = getProms(genes_bed_m)\n",
    "        background_proms_m = getProms(background_genes_bed_m)\n",
    "\n",
    "        proms_m.to_csv(\"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_genes_m_proms.bed\", sep=\"\\t\", header=None, index=None)\n",
    "        background_proms_m.to_csv(\"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_background_genes_m_proms.bed\", sep=\"\\t\", header=None, index=None)\n",
    "\n",
    "        proms_pp = getProms(genes_bed_pp)\n",
    "        background_proms_pp = getProms(background_genes_bed_pp)\n",
    "\n",
    "        proms_pp.to_csv(\"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_genes_pp_proms.bed\", sep=\"\\t\", header=None, index=None)\n",
    "        background_proms_pp.to_csv(\"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_background_genes_pp_proms.bed\", sep=\"\\t\", header=None, index=None)\n",
    "        \n",
    "        proms_pm = getProms(genes_bed_pm)\n",
    "        background_proms_pm = getProms(background_genes_bed_pm)\n",
    "\n",
    "        proms_pm.to_csv(\"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_genes_pm_proms.bed\", sep=\"\\t\", header=None, index=None)\n",
    "        background_proms_pm.to_csv(\"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_background_genes_pm_proms.bed\", sep=\"\\t\", header=None, index=None)\n",
    "        \n",
    "        proms_mp = getProms(genes_bed_mp)\n",
    "        background_proms_mp = getProms(background_genes_bed_mp)\n",
    "\n",
    "        proms_mp.to_csv(\"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_genes_mp_proms.bed\", sep=\"\\t\", header=None, index=None)\n",
    "        background_proms_mp.to_csv(\"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_background_genes_mp_proms.bed\", sep=\"\\t\", header=None, index=None)\n",
    "        \n",
    "        proms_mm = getProms(genes_bed_mm)\n",
    "        background_proms_mm = getProms(background_genes_bed_mm)\n",
    "\n",
    "        proms_mm.to_csv(\"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_genes_mm_proms.bed\", sep=\"\\t\", header=None, index=None)\n",
    "        background_proms_mm.to_csv(\"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_background_genes_mm_proms.bed\", sep=\"\\t\", header=None, index=None)\n",
    "\n",
    "        #for region test\n",
    "        generateBackgroundForRegionTest(target, rna, \"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_background_genes_proms.bed\")\n",
    "        generateBackgroundForRegionTest(target, rna + \"_p\", \"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_background_genes_p_proms.bed\")\n",
    "        generateBackgroundForRegionTest(target, rna + \"_m\", \"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_background_genes_m_proms.bed\")\n",
    "        generateBackgroundForRegionTest(target, rna + \"_pp\", \"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_background_genes_pp_proms.bed\")\n",
    "        generateBackgroundForRegionTest(target, rna + \"_pm\", \"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_background_genes_pm_proms.bed\")\n",
    "        generateBackgroundForRegionTest(target, rna + \"_mp\", \"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_background_genes_mp_proms.bed\")\n",
    "        generateBackgroundForRegionTest(target, rna + \"_mm\", \"../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_background_genes_mm_proms.bed\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    }
   ],
   "source": [
    "generateFilesForTests(\"H3K27ac\", [i[0] for i in rnas_with_prefixes])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getProms(bed):\n",
    "    bed['startProm'] = bed['geneStart'] - 500\n",
    "    bed['startProm'] = bed['startProm'].apply(lambda x: 0 if x < 0 else x)\n",
    "    bed['endProm'] = bed['geneStart'] + 500\n",
    "    \n",
    "    return bed[['cntg', 'startProm', 'endProm', 'geneID', 'geneName', 'strnd']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "def runTDFpromoterTest(rna, prefix):\n",
    "    cmd = \"rgt-TDF promotertest -rm 2 -r ../fantom6/transcripts/\" + rna + \".fa -rn \" + rna + prefix + \" -bed ../H3K27me3/fantom_tdf/\" + rna + \"_genes\" + prefix + \"_proms.bed -bg ../H3K27me3/fantom_tdf/\" + rna + \"_background_genes\" + prefix + \"_proms.bed -o ../H3K27me3/fantom_tdf/promoter_test/ -organism hg19 -l 12 -ccf 20\"\n",
    "    print(cmd)\n",
    "    return subprocess.check_output(cmd, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "def runTDFregionTest(rna, prefix_our, prefix_fantom, target=\"H3K27ac\"):\n",
    "    cmd = \"rgt-TDF regiontest -r ../fantom6/transcripts/\" + rna + \".fa -rn \" + rna + prefix_our + prefix_fantom + \" -bed ../all_marks/\" + target + \"/fantom_tdf/\" + rna + \"_genes\" + prefix_our + prefix_fantom + \"_proms.bed -f ../all_marks/\" + target + \"/fantom_tdf/bg_\" + rna + prefix_our + prefix_fantom + \".bed -o ../all_marks/\" + target + \"/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n \" + str(1000)\n",
    "    print(cmd)\n",
    "    return subprocess.check_output(cmd, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "#bedtools subtract -a A.bed -b B.bed\n",
    "#A - B = substract\n",
    "#геном - промоторы нетаргетных дегов, тогда в качестве бэкграунда будут браться только промоторы нетаргетных дегов(т.е. то, что вычли)\n",
    "def generateBackgroundForRegionTest(target, rna, nontarget_promoters):\n",
    "    proms = BedTool(nontarget_promoters)\n",
    "    hg19 = BedTool('../hg38/chrom.sizes')\n",
    "    hg19.subtract(proms, output='../all_marks/' + target + '/fantom_tdf/bg_' + rna + '.bed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "def runTDF(rna_prefix):\n",
    "    print(runTDFregionTest(rna_prefix[0], rna_prefix[1], rna_prefix[2]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000204054.fa -rn ENSG00000204054 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000204054_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000204054.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000271270.fa -rn ENSG00000271270 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000271270_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000271270.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000268895.fa -rn ENSG00000268895_mm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000268895_genes_mm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000268895_mm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000233396.fa -rn ENSG00000233396 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000233396_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000233396.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000204054.fa -rn ENSG00000204054_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000204054_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000204054_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000204054.fa\\n*** Input regions in BED: ENSG00000204054_genes_pm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000204054_pm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000223485.fa -rn ENSG00000223485_p -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000223485_genes_p_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000223485_p.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000268895.fa\\n*** Input regions in BED: ENSG00000268895_genes_mm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000268895_mm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000214548.fa -rn ENSG00000214548_p -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000214548_genes_p_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000214548_p.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000204054.fa\\n*** Input regions in BED: ENSG00000204054_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000204054\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:19\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:28:10\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:03\\n\\nTotal running time: 0:28:32\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000268895.fa -rn ENSG00000268895 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000268895_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000268895.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000214548.fa\\n*** Input regions in BED: ENSG00000214548_genes_p_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000214548_p\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:28\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:42:00\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:42:30\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000230630.fa -rn ENSG00000230630 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000230630_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000230630.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000271270.fa\\n*** Input regions in BED: ENSG00000271270_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000271270\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:23\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:48:27\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:48:53\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000268895.fa -rn ENSG00000268895_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000268895_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000268895_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000223485.fa\\n*** Input regions in BED: ENSG00000223485_genes_p_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000223485_p\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:36\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 1:13:30\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:03\\n\\nTotal running time: 1:14:08\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000257219.fa -rn ENSG00000257219_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000257219_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000257219_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000257219.fa\\n*** Input regions in BED: ENSG00000257219_genes_pm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000257219_pm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000229647.fa -rn ENSG00000229647_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000229647_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000229647_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000229647.fa\\n*** Input regions in BED: ENSG00000229647_genes_pm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000229647_pm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000214548.fa -rn ENSG00000214548 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000214548_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000214548.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000230630.fa\\n*** Input regions in BED: ENSG00000230630_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000230630\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:13\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:41:52\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:42:08\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000246430.fa -rn ENSG00000246430_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000246430_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000246430_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000246430.fa\\n*** Input regions in BED: ENSG00000246430_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000246430_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:05\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000204054.fa -rn ENSG00000204054_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000204054_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000204054_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000204054.fa\\n*** Input regions in BED: ENSG00000204054_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000204054_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:05\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:18:13\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:18:20\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000240875.fa -rn ENSG00000240875_mm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000240875_genes_mm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000240875_mm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000240875.fa\\n*** Input regions in BED: ENSG00000240875_genes_mm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000240875_mm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000234608.fa -rn ENSG00000234608 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000234608_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000234608.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000268895.fa\\n*** Input regions in BED: ENSG00000268895_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000268895_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:13\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:55:58\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:56:13\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000260032.fa -rn ENSG00000260032_mp -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000260032_genes_mp_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000260032_mp.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000260032.fa\\n*** Input regions in BED: ENSG00000260032_genes_mp_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000260032_mp\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000233396.fa -rn ENSG00000233396_pp -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000233396_genes_pp_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000233396_pp.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000233396.fa\\n*** Input regions in BED: ENSG00000233396_genes_pp_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000233396_pp\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000230630.fa -rn ENSG00000230630_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000230630_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000230630_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000230630.fa\\n*** Input regions in BED: ENSG00000230630_genes_pm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000230630_pm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000227053.fa -rn ENSG00000227053_mm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000227053_genes_mm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000227053_mm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000227053.fa\\n*** Input regions in BED: ENSG00000227053_genes_mm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000227053_mm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000230630.fa -rn ENSG00000230630_mm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000230630_genes_mm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000230630_mm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000230630.fa\\n*** Input regions in BED: ENSG00000230630_genes_mm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000230630_mm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000225733.fa -rn ENSG00000225733_mm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000225733_genes_mm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000225733_mm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000225733.fa\\n*** Input regions in BED: ENSG00000225733_genes_mm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000225733_mm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000223485.fa -rn ENSG00000223485_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000223485_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000223485_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000268895.fa\\n*** Input regions in BED: ENSG00000268895_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000268895\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:19\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 1:24:50\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 1:25:12\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000253161.fa -rn ENSG00000253161_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000253161_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000253161_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000227036.fa -rn ENSG00000227036_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000227036_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000227036_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000227036.fa\\n*** Input regions in BED: ENSG00000227036_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000227036_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:03\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000229852.fa -rn ENSG00000229852_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000229852_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000229852_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000229852.fa\\n*** Input regions in BED: ENSG00000229852_genes_pm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000229852_pm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000230630.fa -rn ENSG00000230630_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000230630_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000230630_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000214548.fa\\n*** Input regions in BED: ENSG00000214548_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000214548\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:10\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 1:01:07\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 1:01:19\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000233396.fa -rn ENSG00000233396_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000233396_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000233396_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000233396.fa\\n*** Input regions in BED: ENSG00000233396_genes_pm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000233396_pm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000268895.fa -rn ENSG00000268895_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000268895_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000268895_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000268895.fa\\n*** Input regions in BED: ENSG00000268895_genes_pm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000268895_pm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000227036.fa -rn ENSG00000227036_mm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000227036_genes_mm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000227036_mm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000227036.fa\\n*** Input regions in BED: ENSG00000227036_genes_mm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000227036_mm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000212978.fa -rn ENSG00000212978 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000212978_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000212978.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000230630.fa\\n*** Input regions in BED: ENSG00000230630_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000230630_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:08\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:23:51\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:24:02\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000260032.fa -rn ENSG00000260032 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000260032_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000260032.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000234608.fa\\n*** Input regions in BED: ENSG00000234608_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000234608\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:07\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:40:44\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:40:53\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000223485.fa -rn ENSG00000223485 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000223485_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000223485.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000212978.fa\\n*** Input regions in BED: ENSG00000212978_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000212978\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:09\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:31:35\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:31:46\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000256268.fa -rn ENSG00000256268_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000256268_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000256268_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000256268.fa\\n*** Input regions in BED: ENSG00000256268_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000256268_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:02\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000212978.fa -rn ENSG00000212978_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000212978_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000212978_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000212978.fa\\n*** Input regions in BED: ENSG00000212978_genes_pm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000212978_pm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000225733.fa -rn ENSG00000225733_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000225733_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000225733_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000233396.fa\\n*** Input regions in BED: ENSG00000233396_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000233396\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:01:41\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 2:47:41\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:04\\n\\nTotal running time: 2:49:26\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000256268.fa -rn ENSG00000256268_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000256268_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000256268_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000256268.fa\\n*** Input regions in BED: ENSG00000256268_genes_pm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000256268_pm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000234608.fa -rn ENSG00000234608_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000234608_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000234608_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000234608.fa\\n*** Input regions in BED: ENSG00000234608_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000234608_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:05\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000246430.fa -rn ENSG00000246430 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000246430_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000246430.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000223485.fa\\n*** Input regions in BED: ENSG00000223485_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000223485_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:15\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 1:18:53\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:03\\n\\nTotal running time: 1:19:11\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000253161.fa -rn ENSG00000253161_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000253161_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000253161_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000271270.fa -rn ENSG00000271270_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000271270_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000271270_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000271270.fa\\n*** Input regions in BED: ENSG00000271270_genes_pm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000271270_pm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000233117.fa -rn ENSG00000233117_mm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000233117_genes_mm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000233117_mm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000233117.fa\\n*** Input regions in BED: ENSG00000233117_genes_mm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000233117_mm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000225733.fa -rn ENSG00000225733_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000225733_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000225733_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000225733.fa\\n*** Input regions in BED: ENSG00000225733_genes_pm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000225733_pm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000227053.fa -rn ENSG00000227053_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000227053_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000227053_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000227053.fa\\n*** Input regions in BED: ENSG00000227053_genes_pm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000227053_pm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000214548.fa -rn ENSG00000214548_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000214548_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000214548_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000214548.fa\\n*** Input regions in BED: ENSG00000214548_genes_pm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000214548_pm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000257219.fa -rn ENSG00000257219_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000257219_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000257219_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000257219.fa\\n*** Input regions in BED: ENSG00000257219_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000257219_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:02\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000229647.fa -rn ENSG00000229647_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000229647_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000229647_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000229647.fa\\n*** Input regions in BED: ENSG00000229647_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000229647_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:02\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000260032.fa -rn ENSG00000260032_p -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000260032_genes_p_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000260032_p.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000225733.fa\\n*** Input regions in BED: ENSG00000225733_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000225733_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:08\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:25:30\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:25:40\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000212978.fa -rn ENSG00000212978_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000212978_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000212978_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000260032.fa\\n*** Input regions in BED: ENSG00000260032_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000260032\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:20\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:55:02\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:55:25\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000253161.fa -rn ENSG00000253161 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000253161_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000253161.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000233396.fa -rn ENSG00000233396_mp -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000233396_genes_mp_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000233396_mp.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000233396.fa\\n*** Input regions in BED: ENSG00000233396_genes_mp_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000233396_mp\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000240875.fa -rn ENSG00000240875_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000240875_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000240875_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000246430.fa\\n*** Input regions in BED: ENSG00000246430_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000246430\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:06\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:30:31\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:30:39\\n'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000223485.fa -rn ENSG00000223485_mm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000223485_genes_mm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000223485_mm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000223485.fa\\n*** Input regions in BED: ENSG00000223485_genes_mm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000223485_mm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000223485.fa -rn ENSG00000223485_pp -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000223485_genes_pp_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000223485_pp.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000223485.fa\\n*** Input regions in BED: ENSG00000223485_genes_pp_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000223485_pp\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000223485.fa -rn ENSG00000223485_mp -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000223485_genes_mp_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000223485_mp.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000223485.fa\\n*** Input regions in BED: ENSG00000223485_genes_mp_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000223485_mp\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000234456.fa -rn ENSG00000234456_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000234456_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000234456_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000234456.fa\\n*** Input regions in BED: ENSG00000234456_genes_pm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000234456_pm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000229852.fa -rn ENSG00000229852_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000229852_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000229852_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000260032.fa\\n*** Input regions in BED: ENSG00000260032_genes_p_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000260032_p\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:09\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:25:07\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:25:19\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000205181.fa -rn ENSG00000205181 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000205181_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000205181.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000205181.fa\\n*** Input regions in BED: ENSG00000205181_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000205181\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:03\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000240875.fa -rn ENSG00000240875 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000240875_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000240875.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000212978.fa\\n*** Input regions in BED: ENSG00000212978_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000212978_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:05\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:20:17\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:20:25\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000256268.fa -rn ENSG00000256268 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000256268_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000256268.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000256268.fa\\n*** Input regions in BED: ENSG00000256268_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000256268\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:03\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000233117.fa -rn ENSG00000233117_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000233117_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000233117_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000233117.fa\\n*** Input regions in BED: ENSG00000233117_genes_pm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000233117_pm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000257219.fa -rn ENSG00000257219 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000257219_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000257219.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000257219.fa\\n*** Input regions in BED: ENSG00000257219_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000257219\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:02\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000229647.fa -rn ENSG00000229647 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000229647_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000229647.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000229647.fa\\n*** Input regions in BED: ENSG00000229647_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000229647\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:02\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000268895.fa -rn ENSG00000268895_p -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000268895_genes_p_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000268895_p.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000240875.fa\\n*** Input regions in BED: ENSG00000240875_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000240875_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:08\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:21:33\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:21:43\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000246430.fa -rn ENSG00000246430_mm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000246430_genes_mm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000246430_mm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000246430.fa\\n*** Input regions in BED: ENSG00000246430_genes_mm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000246430_mm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000233117.fa -rn ENSG00000233117_p -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000233117_genes_p_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000233117_p.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000229852.fa\\n*** Input regions in BED: ENSG00000229852_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000229852_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:04\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:17:20\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:17:25\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000227053.fa -rn ENSG00000227053_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000227053_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000227053_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000227053.fa\\n*** Input regions in BED: ENSG00000227053_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000227053_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:06\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000234456.fa -rn ENSG00000234456_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000234456_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000234456_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000234456.fa\\n*** Input regions in BED: ENSG00000234456_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000234456_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:02\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000246430.fa -rn ENSG00000246430_p -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000246430_genes_p_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000246430_p.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000246430.fa\\n*** Input regions in BED: ENSG00000246430_genes_p_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000246430_p\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:04\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000230630.fa -rn ENSG00000230630_mp -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000230630_genes_mp_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000230630_mp.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000230630.fa\\n*** Input regions in BED: ENSG00000230630_genes_mp_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000230630_mp\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000271270.fa -rn ENSG00000271270_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000271270_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000271270_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000233117.fa\\n*** Input regions in BED: ENSG00000233117_genes_p_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000233117_p\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:05\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:18:20\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:18:27\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000225733.fa -rn ENSG00000225733 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000225733_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000225733.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000271270.fa\\n*** Input regions in BED: ENSG00000271270_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000271270_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:10\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:24:45\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:24:57\\n'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000229647.fa -rn ENSG00000229647_mm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000229647_genes_mm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000229647_mm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000229647.fa\\n*** Input regions in BED: ENSG00000229647_genes_mm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000229647_mm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000234608.fa -rn ENSG00000234608_mm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000234608_genes_mm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000234608_mm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000234608.fa\\n*** Input regions in BED: ENSG00000234608_genes_mm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000234608_mm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000214548.fa -rn ENSG00000214548_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000214548_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000214548_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000214548.fa\\n*** Input regions in BED: ENSG00000214548_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000214548_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:05\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000233117.fa -rn ENSG00000233117_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000233117_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000233117_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000240875.fa\\n*** Input regions in BED: ENSG00000240875_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000240875\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:12\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:32:52\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:33:06\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000233117.fa -rn ENSG00000233117_mp -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000233117_genes_mp_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000233117_mp.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000233117.fa\\n*** Input regions in BED: ENSG00000233117_genes_mp_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000233117_mp\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000233396.fa -rn ENSG00000233396_m -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000233396_genes_m_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000233396_m.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000268895.fa\\n*** Input regions in BED: ENSG00000268895_genes_p_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000268895_p\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:09\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:33:17\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:33:28\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000234456.fa -rn ENSG00000234456 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000234456_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000234456.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000234456.fa\\n*** Input regions in BED: ENSG00000234456_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000234456\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:02\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000268895.fa -rn ENSG00000268895_mp -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000268895_genes_mp_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000268895_mp.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000268895.fa\\n*** Input regions in BED: ENSG00000268895_genes_mp_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000268895_mp\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000233117.fa -rn ENSG00000233117 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000233117_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000233117.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000233117.fa\\n*** Input regions in BED: ENSG00000233117_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000233117_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:03\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:14:21\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:14:25\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000246430.fa -rn ENSG00000246430_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000246430_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000246430_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000246430.fa\\n*** Input regions in BED: ENSG00000246430_genes_pm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000246430_pm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000229852.fa -rn ENSG00000229852 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000229852_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000229852.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000223485.fa\\n*** Input regions in BED: ENSG00000223485_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000223485\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:27\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 1:53:08\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 1:53:36\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000246430.fa -rn ENSG00000246430_mp -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000246430_genes_mp_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000246430_mp.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000246430.fa\\n*** Input regions in BED: ENSG00000246430_genes_mp_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000246430_mp\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000223485.fa -rn ENSG00000223485_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000223485_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000223485_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000223485.fa\\n*** Input regions in BED: ENSG00000223485_genes_pm_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000223485_pm\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000225733.fa\\n*** Input regions in BED: ENSG00000225733_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000225733\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:06\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:28:29\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:01\\n\\nTotal running time: 0:28:36\\n'\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000233117.fa\\n*** Input regions in BED: ENSG00000233117_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000233117\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:07\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:24:01\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:01\\n\\nTotal running time: 0:24:09\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000214548.fa -rn ENSG00000214548_mp -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000214548_genes_mp_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000214548_mp.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000214548.fa\\n*** Input regions in BED: ENSG00000214548_genes_mp_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000214548_mp\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:00\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n*** Find no DBD having DBS with cutoff = 50\\n'\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000229852.fa\\n*** Input regions in BED: ENSG00000229852_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000229852\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:04\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:15:29\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:01\\n\\nTotal running time: 0:15:35\\n'\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000233396.fa\\n*** Input regions in BED: ENSG00000233396_genes_m_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000233396_m\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:20\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:51:10\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:51:32\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000230630.fa -rn ENSG00000230630_p -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000230630_genes_p_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000230630_p.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000230630.fa\\n*** Input regions in BED: ENSG00000230630_genes_p_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000230630_p\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:03\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:07:36\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:01\\n\\nTotal running time: 0:07:41\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000233396.fa -rn ENSG00000233396_p -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000233396_genes_p_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000233396_p.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n",
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000233396.fa\\n*** Input regions in BED: ENSG00000233396_genes_p_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000233396_p\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:14\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:33:37\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:02\\n\\nTotal running time: 0:33:52\\n'\n",
      "rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000227053.fa -rn ENSG00000227053 -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000227053_genes_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000227053.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "b'\\n*************** Genomic Region Test ***************\\n*** Input RNA sequence: /home/mazurovev/fantom6/transcripts/ENSG00000227053.fa\\n*** Input regions in BED: ENSG00000227053_genes_proms.bed\\n*** Number of randomization: 1000\\n*** Output directory: ENSG00000227053\\nStep 1: Calculate the triplex forming sites on RNA and DNA.\\n\\tRunning time: 0:00:05\\nStep 2: Permutation by randomization the target regions for 1000 times.\\n\\t\\t|0%                  |                100%|\\n\\t\\t[=========================================]\\n\\tRunning time: 0:21:06\\nStep 3: Generate plot and output html files.\\n\\tRunning time: 0:00:01\\n\\nTotal running time: 0:21:12\\n'\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000253161.fa -rn ENSG00000253161_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000253161_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000253161_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000' returned non-zero exit status 1.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mRemoteTraceback\u001b[0m                           Traceback (most recent call last)",
      "\u001b[0;31mRemoteTraceback\u001b[0m: \n\"\"\"\nTraceback (most recent call last):\n  File \"/usr/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n    result = (True, func(*args, **kwds))\n  File \"/usr/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n    return list(map(*args))\n  File \"<ipython-input-28-63d6427d815d>\", line 2, in runTDF\n    print(runTDFregionTest(rna_prefix[0], rna_prefix[1], rna_prefix[2]))\n  File \"<ipython-input-34-b03c97ad77f7>\", line 4, in runTDFregionTest\n    return subprocess.check_output(cmd, shell=True)\n  File \"/usr/lib/python3.6/subprocess.py\", line 356, in check_output\n    **kwargs).stdout\n  File \"/usr/lib/python3.6/subprocess.py\", line 438, in run\n    output=stdout, stderr=stderr)\nsubprocess.CalledProcessError: Command 'rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000253161.fa -rn ENSG00000253161_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000253161_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000253161_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000' returned non-zero exit status 1.\n\"\"\"",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-35-578006cb8dac>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mpool\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mPool\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprocesses\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m5\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpool\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmap\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrunTDF\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrnas_with_prefixes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mpool\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mpool\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.6/multiprocessing/pool.py\u001b[0m in \u001b[0;36mmap\u001b[0;34m(self, func, iterable, chunksize)\u001b[0m\n\u001b[1;32m    264\u001b[0m         \u001b[0;32min\u001b[0m \u001b[0ma\u001b[0m \u001b[0mlist\u001b[0m \u001b[0mthat\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0mreturned\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    265\u001b[0m         '''\n\u001b[0;32m--> 266\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_map_async\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfunc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0miterable\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmapstar\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mchunksize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    267\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    268\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mstarmap\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0miterable\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mchunksize\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.6/multiprocessing/pool.py\u001b[0m in \u001b[0;36mget\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    642\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_value\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    643\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 644\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_value\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    645\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    646\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_set\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mobj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mCalledProcessError\u001b[0m: Command 'rgt-TDF regiontest -r ../fantom6/transcripts/ENSG00000253161.fa -rn ENSG00000253161_pm -bed ../all_marks/H3K27ac/fantom_tdf/ENSG00000253161_genes_pm_proms.bed -f ../all_marks/H3K27ac/fantom_tdf/bg_ENSG00000253161_pm.bed -o ../all_marks/H3K27ac/fantom_tdf/region_test/ -organism hg38 -l 12 -e 10 -obed -rt -mp 6 -ccf 50 -n1000' returned non-zero exit status 1."
     ]
    }
   ],
   "source": [
    "pool = Pool(processes=5)\n",
    "res = pool.map(runTDF, rnas_with_prefixes)\n",
    "\n",
    "pool.close()\n",
    "pool.join()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "def tdfDataToHtml(target):\n",
    "    import os\n",
    "    folders = os.listdir(\"../all_marks/\" + target + \"/fantom_tdf/region_test\")\n",
    "\n",
    "    for path in folders:\n",
    "        print(path)\n",
    "        try:\n",
    "            #os.chmod(\"../H3K27me3/fantom_tdf/region_test/\" + path + \"/data.tsv\", 0o777)\n",
    "            getHtmlTableWithTDFResults(\"../all_marks/\" + target + \"/fantom_tdf/region_test/\" + path)\n",
    "        except FileNotFoundError: \n",
    "            print('Oops, ' + path + '/data.tsv not found')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getHtmlTableWithTDFResults(path):\n",
    "    def color_negative_red(row):\n",
    "        val = row[5]\n",
    "        try:\n",
    "            color = 'red' if val < 0.05 else 'black'\n",
    "        except Exception:\n",
    "            color = 'black'\n",
    "        return ['color: black']*5 + ['color: %s' % color] + ['color: black']\n",
    "    \n",
    "    def processTable(df):\n",
    "        new = df['DBD'].str.split(\"-\", n = 1, expand = True) \n",
    "        new[0] = pd.to_numeric(new[0])\n",
    "        new[1] = pd.to_numeric(new[1])\n",
    "        \n",
    "        df['startDBD'] = new[0]\n",
    "        df['endDBD'] = new[1]\n",
    "        \n",
    "        df.sort_values(by=['startDBD'], ascending=True, inplace=True)\n",
    "        df.drop(columns=['startDBD', 'endDBD'], inplace=True)\n",
    "        df = df[df['p-value'] < 0.05]\n",
    "        df = df.reset_index(drop=True)\n",
    "        \n",
    "        return df\n",
    "    \n",
    "    df = processTable(pd.read_csv(path + \"/data.tsv\", sep=\"\\t\"))\n",
    "    #s = df.style.apply(color_negative_red, axis=1)\n",
    "    s = df.style\n",
    "    \n",
    "    with open(path + \"/data_table.html\", 'w') as f:\n",
    "        for index, item in enumerate(s.render().split(\"\\n\")):\n",
    "            if(index == 0):\n",
    "                f.write('<style  type=\"text/css\" >\\n')\n",
    "                f.write('\\n')\n",
    "                f.write('table, th, td {\\n')\n",
    "                f.write('border: 1px solid black;\\n')\n",
    "                f.write('border-collapse: collapse;\\n')\n",
    "                f.write('font-size: 11px;\\n')\n",
    "                f.write('}\\n')\n",
    "                f.write('\\n')\n",
    "                f.write('th,\\n')\n",
    "                f.write('td {\\n')\n",
    "                f.write('border: 1px solid black;\\n')\n",
    "                f.write('width: 100px;\\n')\n",
    "                f.write('height: 25px;\\n')\n",
    "                f.write('text-align:center;\\n')\n",
    "                f.write('font-family: Montserrat;\\n')\n",
    "                f.write('overflow: hidden;\\n')\n",
    "                f.write('}\\n')\n",
    "                f.write('\\n')\n",
    "                f.write('tr:nth-child(even) {\\n')\n",
    "                f.write('background-color: #ffe6e6\\n')\n",
    "                f.write('}\\n')\n",
    "                f.write('\\n')\n",
    "            else:\n",
    "                f.write(\"%s\\n\" % item)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "def processTable(df):\n",
    "        new = df['DBD'].str.split(\"-\", n = 1, expand = True) \n",
    "        new[0] = pd.to_numeric(new[0])\n",
    "        new[1] = pd.to_numeric(new[1])\n",
    "        \n",
    "        df['startDBD'] = new[0]\n",
    "        df['endDBD'] = new[1]\n",
    "        \n",
    "        df.sort_values(by=['startDBD'], ascending=True, inplace=True)\n",
    "        df.drop(columns=['startDBD', 'endDBD'], inplace=True)\n",
    "        df = df[df['p-value'] < 0.05]\n",
    "        df = df.reset_index(drop=True)\n",
    "        \n",
    "        return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENSG00000234608_m\n",
      "Oops, ENSG00000234608_m/data.tsv not found\n",
      "ENSG00000256268_m\n",
      "Oops, ENSG00000256268_m/data.tsv not found\n",
      "ENSG00000229647\n",
      "Oops, ENSG00000229647/data.tsv not found\n",
      "ENSG00000233117_mm\n",
      "Oops, ENSG00000233117_mm/data.tsv not found\n",
      "ENSG00000233117_p\n",
      "ENSG00000268895_p\n",
      "ENSG00000223485_p\n",
      "ENSG00000212978_pm\n",
      "Oops, ENSG00000212978_pm/data.tsv not found\n",
      "ENSG00000229852\n",
      "ENSG00000257219_m\n",
      "Oops, ENSG00000257219_m/data.tsv not found\n",
      "ENSG00000204054_pm\n",
      "Oops, ENSG00000204054_pm/data.tsv not found\n",
      "ENSG00000246430\n",
      "ENSG00000240875_mm\n",
      "Oops, ENSG00000240875_mm/data.tsv not found\n",
      "ENSG00000268895_m\n",
      "ENSG00000229852_m\n",
      "ENSG00000227053\n",
      "ENSG00000223485_mm\n",
      "Oops, ENSG00000223485_mm/data.tsv not found\n",
      "ENSG00000227053_m\n",
      "Oops, ENSG00000227053_m/data.tsv not found\n",
      "ENSG00000230630_mm\n",
      "Oops, ENSG00000230630_mm/data.tsv not found\n",
      "ENSG00000214548_mp\n",
      "Oops, ENSG00000214548_mp/data.tsv not found\n",
      "ENSG00000227053_mm\n",
      "Oops, ENSG00000227053_mm/data.tsv not found\n",
      "ENSG00000256268\n",
      "Oops, ENSG00000256268/data.tsv not found\n",
      "ENSG00000246430_mm\n",
      "Oops, ENSG00000246430_mm/data.tsv not found\n",
      "ENSG00000271270_pm\n",
      "Oops, ENSG00000271270_pm/data.tsv not found\n",
      "ENSG00000233396_p\n",
      "ENSG00000229647_pm\n",
      "Oops, ENSG00000229647_pm/data.tsv not found\n",
      "ENSG00000214548_p\n",
      "ENSG00000227053_pm\n",
      "Oops, ENSG00000227053_pm/data.tsv not found\n",
      "ENSG00000212978\n",
      "ENSG00000230630_p\n",
      "ENSG00000225733_m\n",
      "ENSG00000230630_pm\n",
      "Oops, ENSG00000230630_pm/data.tsv not found\n",
      "ENSG00000260032_p\n",
      "ENSG00000268895\n",
      "ENSG00000230630_mp\n",
      "Oops, ENSG00000230630_mp/data.tsv not found\n",
      "ENSG00000260032\n",
      "ENSG00000227036_mm\n",
      "Oops, ENSG00000227036_mm/data.tsv not found\n",
      "ENSG00000246430_m\n",
      "Oops, ENSG00000246430_m/data.tsv not found\n",
      "ENSG00000212978_m\n",
      "ENSG00000233396_pp\n",
      "Oops, ENSG00000233396_pp/data.tsv not found\n",
      "ENSG00000227036_m\n",
      "Oops, ENSG00000227036_m/data.tsv not found\n",
      "ENSG00000223485_pm\n",
      "Oops, ENSG00000223485_pm/data.tsv not found\n",
      "ENSG00000204054_m\n",
      "ENSG00000268895_mp\n",
      "Oops, ENSG00000268895_mp/data.tsv not found\n",
      "ENSG00000233396\n",
      "ENSG00000230630\n",
      "ENSG00000225733_mm\n",
      "Oops, ENSG00000225733_mm/data.tsv not found\n",
      "ENSG00000246430_mp\n",
      "Oops, ENSG00000246430_mp/data.tsv not found\n",
      "ENSG00000229647_m\n",
      "Oops, ENSG00000229647_m/data.tsv not found\n",
      "ENSG00000233396_pm\n",
      "Oops, ENSG00000233396_pm/data.tsv not found\n",
      "ENSG00000253161\n",
      "Oops, ENSG00000253161/data.tsv not found\n",
      "ENSG00000214548_m\n",
      "Oops, ENSG00000214548_m/data.tsv not found\n",
      "style\n",
      "Oops, style/data.tsv not found\n",
      "ENSG00000260032_mp\n",
      "Oops, ENSG00000260032_mp/data.tsv not found\n",
      "ENSG00000268895_pm\n",
      "Oops, ENSG00000268895_pm/data.tsv not found\n",
      "ENSG00000223485_mp\n",
      "Oops, ENSG00000223485_mp/data.tsv not found\n",
      "ENSG00000223485_m\n",
      "ENSG00000257219_pm\n",
      "Oops, ENSG00000257219_pm/data.tsv not found\n",
      "ENSG00000233117_pm\n",
      "Oops, ENSG00000233117_pm/data.tsv not found\n",
      "ENSG00000233117_m\n",
      "ENSG00000268895_mm\n",
      "Oops, ENSG00000268895_mm/data.tsv not found\n",
      "ENSG00000246430_p\n",
      "Oops, ENSG00000246430_p/data.tsv not found\n",
      "ENSG00000205181\n",
      "Oops, ENSG00000205181/data.tsv not found\n",
      "ENSG00000225733\n",
      "ENSG00000214548\n",
      "ENSG00000233396_mp\n",
      "Oops, ENSG00000233396_mp/data.tsv not found\n",
      "ENSG00000233396_m\n",
      "ENSG00000234456\n",
      "Oops, ENSG00000234456/data.tsv not found\n",
      "ENSG00000229647_mm\n",
      "Oops, ENSG00000229647_mm/data.tsv not found\n",
      "ENSG00000246430_pm\n",
      "Oops, ENSG00000246430_pm/data.tsv not found\n",
      "ENSG00000271270\n",
      "ENSG00000234456_pm\n",
      "Oops, ENSG00000234456_pm/data.tsv not found\n",
      "ENSG00000234608\n",
      "ENSG00000225733_pm\n",
      "Oops, ENSG00000225733_pm/data.tsv not found\n",
      "ENSG00000223485_pp\n",
      "Oops, ENSG00000223485_pp/data.tsv not found\n",
      "ENSG00000253161_m\n",
      "Oops, ENSG00000253161_m/data.tsv not found\n",
      "ENSG00000240875_m\n",
      "ENSG00000204054\n",
      "ENSG00000257219\n",
      "Oops, ENSG00000257219/data.tsv not found\n",
      "ENSG00000230630_m\n",
      "ENSG00000233117\n",
      "ENSG00000240875\n",
      "ENSG00000256268_pm\n",
      "Oops, ENSG00000256268_pm/data.tsv not found\n",
      "ENSG00000233117_mp\n",
      "Oops, ENSG00000233117_mp/data.tsv not found\n",
      "ENSG00000234608_mm\n",
      "Oops, ENSG00000234608_mm/data.tsv not found\n",
      "ENSG00000234456_m\n",
      "Oops, ENSG00000234456_m/data.tsv not found\n",
      "ENSG00000253161_pm\n",
      "Oops, ENSG00000253161_pm/data.tsv not found\n",
      "ENSG00000229852_pm\n",
      "Oops, ENSG00000229852_pm/data.tsv not found\n",
      "ENSG00000223485\n",
      "ENSG00000271270_m\n",
      "ENSG00000214548_pm\n",
      "Oops, ENSG00000214548_pm/data.tsv not found\n"
     ]
    }
   ],
   "source": [
    "tdfDataToHtml(\"H3K27ac\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "h3k27ac_ccf_50_res = ['ENSG00000233117_p','ENSG00000268895_p','ENSG00000223485_p','ENSG00000229852',\n",
    "                      'ENSG00000246430','ENSG00000268895_m','ENSG00000229852_m','ENSG00000227053',\n",
    "                      'ENSG00000233396_p','ENSG00000214548_p','ENSG00000212978','ENSG00000230630_p',\n",
    "                      'ENSG00000225733_m','ENSG00000260032_p','ENSG00000268895','ENSG00000260032',\n",
    "                      'ENSG00000212978_m','ENSG00000204054_m','ENSG00000233396','ENSG00000230630',\n",
    "                      'ENSG00000223485_m','ENSG00000233117_m','ENSG00000225733','ENSG00000214548',\n",
    "                      'ENSG00000233396_m','ENSG00000271270','ENSG00000234608','ENSG00000240875_m',\n",
    "                      'ENSG00000204054','ENSG00000230630_m','ENSG00000233117','ENSG00000240875',\n",
    "                      'ENSG00000223485','ENSG00000271270_m']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "res = []\n",
    "for f in h3k27ac_ccf_50_res:\n",
    "    tmp = pd.read_csv(\"../all_marks/H3K27ac/fantom_tdf/region_test/\" + f + \"/data.tsv\", sep=\"\\t\")\n",
    "    tmp = tmp[tmp['p-value'] < 0.05]\n",
    "    if(tmp.shape[0] > 0):\n",
    "        tmp['lnc'] = [f]*tmp.shape[0]\n",
    "        res.append(tmp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "res = pd.concat(res, ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DBD</th>\n",
       "      <th>tr_with DBS</th>\n",
       "      <th>tr_without DBS</th>\n",
       "      <th>ntr_with DBS(average)</th>\n",
       "      <th>ntr_with DBS(std)</th>\n",
       "      <th>p-value</th>\n",
       "      <th>z-score</th>\n",
       "      <th>lnc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>2406-2425</td>\n",
       "      <td>68</td>\n",
       "      <td>882</td>\n",
       "      <td>45.1</td>\n",
       "      <td>6.29</td>\n",
       "      <td>0.000</td>\n",
       "      <td>3.63</td>\n",
       "      <td>ENSG00000212978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>1537-1558</td>\n",
       "      <td>73</td>\n",
       "      <td>877</td>\n",
       "      <td>58.9</td>\n",
       "      <td>7.25</td>\n",
       "      <td>0.024</td>\n",
       "      <td>1.94</td>\n",
       "      <td>ENSG00000212978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>1980-2011</td>\n",
       "      <td>289</td>\n",
       "      <td>661</td>\n",
       "      <td>250.2</td>\n",
       "      <td>13.40</td>\n",
       "      <td>0.001</td>\n",
       "      <td>2.88</td>\n",
       "      <td>ENSG00000212978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>9-28</td>\n",
       "      <td>79</td>\n",
       "      <td>3084</td>\n",
       "      <td>63.2</td>\n",
       "      <td>7.09</td>\n",
       "      <td>0.012</td>\n",
       "      <td>2.24</td>\n",
       "      <td>ENSG00000214548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>9-28</td>\n",
       "      <td>51</td>\n",
       "      <td>1883</td>\n",
       "      <td>40.9</td>\n",
       "      <td>5.86</td>\n",
       "      <td>0.031</td>\n",
       "      <td>1.72</td>\n",
       "      <td>ENSG00000214548_p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>84-96</td>\n",
       "      <td>724</td>\n",
       "      <td>5848</td>\n",
       "      <td>496.3</td>\n",
       "      <td>15.80</td>\n",
       "      <td>0.000</td>\n",
       "      <td>14.40</td>\n",
       "      <td>ENSG00000223485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>108-129</td>\n",
       "      <td>108</td>\n",
       "      <td>6464</td>\n",
       "      <td>85.2</td>\n",
       "      <td>6.75</td>\n",
       "      <td>0.000</td>\n",
       "      <td>3.39</td>\n",
       "      <td>ENSG00000223485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>240-252</td>\n",
       "      <td>68</td>\n",
       "      <td>6504</td>\n",
       "      <td>43.0</td>\n",
       "      <td>4.81</td>\n",
       "      <td>0.000</td>\n",
       "      <td>5.20</td>\n",
       "      <td>ENSG00000223485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>339-356</td>\n",
       "      <td>75</td>\n",
       "      <td>6497</td>\n",
       "      <td>56.0</td>\n",
       "      <td>5.56</td>\n",
       "      <td>0.000</td>\n",
       "      <td>3.42</td>\n",
       "      <td>ENSG00000223485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>733-749</td>\n",
       "      <td>155</td>\n",
       "      <td>6417</td>\n",
       "      <td>113.0</td>\n",
       "      <td>7.79</td>\n",
       "      <td>0.000</td>\n",
       "      <td>5.40</td>\n",
       "      <td>ENSG00000223485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>178-195</td>\n",
       "      <td>136</td>\n",
       "      <td>6436</td>\n",
       "      <td>104.9</td>\n",
       "      <td>7.38</td>\n",
       "      <td>0.000</td>\n",
       "      <td>4.21</td>\n",
       "      <td>ENSG00000223485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>920-937</td>\n",
       "      <td>146</td>\n",
       "      <td>6426</td>\n",
       "      <td>102.9</td>\n",
       "      <td>7.22</td>\n",
       "      <td>0.000</td>\n",
       "      <td>5.98</td>\n",
       "      <td>ENSG00000223485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>1019-1040</td>\n",
       "      <td>190</td>\n",
       "      <td>6382</td>\n",
       "      <td>140.7</td>\n",
       "      <td>8.49</td>\n",
       "      <td>0.000</td>\n",
       "      <td>5.81</td>\n",
       "      <td>ENSG00000223485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>955-972</td>\n",
       "      <td>248</td>\n",
       "      <td>6324</td>\n",
       "      <td>192.3</td>\n",
       "      <td>9.78</td>\n",
       "      <td>0.000</td>\n",
       "      <td>5.70</td>\n",
       "      <td>ENSG00000223485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>901-918</td>\n",
       "      <td>210</td>\n",
       "      <td>6362</td>\n",
       "      <td>155.4</td>\n",
       "      <td>9.38</td>\n",
       "      <td>0.000</td>\n",
       "      <td>5.82</td>\n",
       "      <td>ENSG00000223485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>84-96</td>\n",
       "      <td>425</td>\n",
       "      <td>3251</td>\n",
       "      <td>327.6</td>\n",
       "      <td>15.00</td>\n",
       "      <td>0.000</td>\n",
       "      <td>6.49</td>\n",
       "      <td>ENSG00000223485_m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>955-972</td>\n",
       "      <td>142</td>\n",
       "      <td>3534</td>\n",
       "      <td>124.6</td>\n",
       "      <td>9.44</td>\n",
       "      <td>0.031</td>\n",
       "      <td>1.84</td>\n",
       "      <td>ENSG00000223485_m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>920-937</td>\n",
       "      <td>90</td>\n",
       "      <td>3586</td>\n",
       "      <td>66.3</td>\n",
       "      <td>7.30</td>\n",
       "      <td>0.000</td>\n",
       "      <td>3.25</td>\n",
       "      <td>ENSG00000223485_m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>733-749</td>\n",
       "      <td>94</td>\n",
       "      <td>3582</td>\n",
       "      <td>73.4</td>\n",
       "      <td>7.20</td>\n",
       "      <td>0.001</td>\n",
       "      <td>2.87</td>\n",
       "      <td>ENSG00000223485_m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>1019-1040</td>\n",
       "      <td>114</td>\n",
       "      <td>3562</td>\n",
       "      <td>89.3</td>\n",
       "      <td>8.19</td>\n",
       "      <td>0.000</td>\n",
       "      <td>3.02</td>\n",
       "      <td>ENSG00000223485_m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>108-129</td>\n",
       "      <td>67</td>\n",
       "      <td>3609</td>\n",
       "      <td>53.7</td>\n",
       "      <td>6.40</td>\n",
       "      <td>0.009</td>\n",
       "      <td>2.08</td>\n",
       "      <td>ENSG00000223485_m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>901-918</td>\n",
       "      <td>123</td>\n",
       "      <td>3192</td>\n",
       "      <td>89.5</td>\n",
       "      <td>8.32</td>\n",
       "      <td>0.000</td>\n",
       "      <td>4.03</td>\n",
       "      <td>ENSG00000223485_p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>84-96</td>\n",
       "      <td>337</td>\n",
       "      <td>2978</td>\n",
       "      <td>310.8</td>\n",
       "      <td>15.20</td>\n",
       "      <td>0.036</td>\n",
       "      <td>1.72</td>\n",
       "      <td>ENSG00000223485_p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>2539-2558</td>\n",
       "      <td>223</td>\n",
       "      <td>1004</td>\n",
       "      <td>200.7</td>\n",
       "      <td>12.60</td>\n",
       "      <td>0.034</td>\n",
       "      <td>1.77</td>\n",
       "      <td>ENSG00000225733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>2433-2446</td>\n",
       "      <td>59</td>\n",
       "      <td>1168</td>\n",
       "      <td>44.0</td>\n",
       "      <td>6.30</td>\n",
       "      <td>0.009</td>\n",
       "      <td>2.38</td>\n",
       "      <td>ENSG00000225733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>681-693</td>\n",
       "      <td>86</td>\n",
       "      <td>1001</td>\n",
       "      <td>69.0</td>\n",
       "      <td>7.79</td>\n",
       "      <td>0.009</td>\n",
       "      <td>2.19</td>\n",
       "      <td>ENSG00000229852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1052-1067</td>\n",
       "      <td>80</td>\n",
       "      <td>1007</td>\n",
       "      <td>62.4</td>\n",
       "      <td>7.34</td>\n",
       "      <td>0.014</td>\n",
       "      <td>2.40</td>\n",
       "      <td>ENSG00000229852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>116-144</td>\n",
       "      <td>73</td>\n",
       "      <td>1014</td>\n",
       "      <td>55.9</td>\n",
       "      <td>6.81</td>\n",
       "      <td>0.002</td>\n",
       "      <td>2.50</td>\n",
       "      <td>ENSG00000229852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>681-693</td>\n",
       "      <td>55</td>\n",
       "      <td>607</td>\n",
       "      <td>42.5</td>\n",
       "      <td>6.15</td>\n",
       "      <td>0.018</td>\n",
       "      <td>2.04</td>\n",
       "      <td>ENSG00000229852_m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>2513-2588</td>\n",
       "      <td>401</td>\n",
       "      <td>723</td>\n",
       "      <td>363.2</td>\n",
       "      <td>14.90</td>\n",
       "      <td>0.005</td>\n",
       "      <td>2.54</td>\n",
       "      <td>ENSG00000230630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>46-94</td>\n",
       "      <td>1186</td>\n",
       "      <td>6147</td>\n",
       "      <td>827.2</td>\n",
       "      <td>18.30</td>\n",
       "      <td>0.000</td>\n",
       "      <td>19.60</td>\n",
       "      <td>ENSG00000233396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>1153-1174</td>\n",
       "      <td>237</td>\n",
       "      <td>3685</td>\n",
       "      <td>189.9</td>\n",
       "      <td>12.10</td>\n",
       "      <td>0.000</td>\n",
       "      <td>3.91</td>\n",
       "      <td>ENSG00000233396_m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>1079-1100</td>\n",
       "      <td>568</td>\n",
       "      <td>3354</td>\n",
       "      <td>509.5</td>\n",
       "      <td>18.00</td>\n",
       "      <td>0.000</td>\n",
       "      <td>3.24</td>\n",
       "      <td>ENSG00000233396_m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>1904-1923</td>\n",
       "      <td>178</td>\n",
       "      <td>3744</td>\n",
       "      <td>160.4</td>\n",
       "      <td>10.60</td>\n",
       "      <td>0.044</td>\n",
       "      <td>1.65</td>\n",
       "      <td>ENSG00000233396_m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>521-541</td>\n",
       "      <td>276</td>\n",
       "      <td>3646</td>\n",
       "      <td>237.6</td>\n",
       "      <td>12.50</td>\n",
       "      <td>0.000</td>\n",
       "      <td>3.07</td>\n",
       "      <td>ENSG00000233396_m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>1428-1444</td>\n",
       "      <td>162</td>\n",
       "      <td>3760</td>\n",
       "      <td>135.0</td>\n",
       "      <td>9.71</td>\n",
       "      <td>0.002</td>\n",
       "      <td>2.78</td>\n",
       "      <td>ENSG00000233396_m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>46-94</td>\n",
       "      <td>619</td>\n",
       "      <td>3303</td>\n",
       "      <td>549.1</td>\n",
       "      <td>18.60</td>\n",
       "      <td>0.001</td>\n",
       "      <td>3.76</td>\n",
       "      <td>ENSG00000233396_m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>1732-1744</td>\n",
       "      <td>90</td>\n",
       "      <td>3832</td>\n",
       "      <td>77.1</td>\n",
       "      <td>7.63</td>\n",
       "      <td>0.043</td>\n",
       "      <td>1.70</td>\n",
       "      <td>ENSG00000233396_m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>1667-1680</td>\n",
       "      <td>60</td>\n",
       "      <td>3361</td>\n",
       "      <td>48.2</td>\n",
       "      <td>6.16</td>\n",
       "      <td>0.025</td>\n",
       "      <td>1.92</td>\n",
       "      <td>ENSG00000233396_p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>1904-1923</td>\n",
       "      <td>159</td>\n",
       "      <td>3262</td>\n",
       "      <td>141.8</td>\n",
       "      <td>10.20</td>\n",
       "      <td>0.035</td>\n",
       "      <td>1.68</td>\n",
       "      <td>ENSG00000233396_p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>1732-1744</td>\n",
       "      <td>81</td>\n",
       "      <td>3340</td>\n",
       "      <td>68.6</td>\n",
       "      <td>7.23</td>\n",
       "      <td>0.049</td>\n",
       "      <td>1.71</td>\n",
       "      <td>ENSG00000233396_p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>46-94</td>\n",
       "      <td>564</td>\n",
       "      <td>2857</td>\n",
       "      <td>482.1</td>\n",
       "      <td>17.90</td>\n",
       "      <td>0.000</td>\n",
       "      <td>4.57</td>\n",
       "      <td>ENSG00000233396_p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>1079-1100</td>\n",
       "      <td>524</td>\n",
       "      <td>2897</td>\n",
       "      <td>447.5</td>\n",
       "      <td>17.80</td>\n",
       "      <td>0.000</td>\n",
       "      <td>4.29</td>\n",
       "      <td>ENSG00000233396_p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>1825-1846</td>\n",
       "      <td>161</td>\n",
       "      <td>3260</td>\n",
       "      <td>117.0</td>\n",
       "      <td>9.39</td>\n",
       "      <td>0.000</td>\n",
       "      <td>4.69</td>\n",
       "      <td>ENSG00000233396_p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1428-1444</td>\n",
       "      <td>146</td>\n",
       "      <td>3275</td>\n",
       "      <td>119.4</td>\n",
       "      <td>9.42</td>\n",
       "      <td>0.003</td>\n",
       "      <td>2.82</td>\n",
       "      <td>ENSG00000233396_p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>1153-1174</td>\n",
       "      <td>196</td>\n",
       "      <td>3225</td>\n",
       "      <td>172.1</td>\n",
       "      <td>11.50</td>\n",
       "      <td>0.020</td>\n",
       "      <td>2.09</td>\n",
       "      <td>ENSG00000233396_p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>521-541</td>\n",
       "      <td>244</td>\n",
       "      <td>3177</td>\n",
       "      <td>212.4</td>\n",
       "      <td>12.20</td>\n",
       "      <td>0.005</td>\n",
       "      <td>2.58</td>\n",
       "      <td>ENSG00000233396_p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>1509-1535</td>\n",
       "      <td>130</td>\n",
       "      <td>1008</td>\n",
       "      <td>104.4</td>\n",
       "      <td>9.57</td>\n",
       "      <td>0.003</td>\n",
       "      <td>2.67</td>\n",
       "      <td>ENSG00000240875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>440-457</td>\n",
       "      <td>97</td>\n",
       "      <td>1041</td>\n",
       "      <td>77.9</td>\n",
       "      <td>8.16</td>\n",
       "      <td>0.013</td>\n",
       "      <td>2.34</td>\n",
       "      <td>ENSG00000240875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>694-713</td>\n",
       "      <td>162</td>\n",
       "      <td>976</td>\n",
       "      <td>130.7</td>\n",
       "      <td>10.30</td>\n",
       "      <td>0.000</td>\n",
       "      <td>3.04</td>\n",
       "      <td>ENSG00000240875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>2259-2290</td>\n",
       "      <td>253</td>\n",
       "      <td>930</td>\n",
       "      <td>231.3</td>\n",
       "      <td>13.30</td>\n",
       "      <td>0.043</td>\n",
       "      <td>1.63</td>\n",
       "      <td>ENSG00000260032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>130-150</td>\n",
       "      <td>255</td>\n",
       "      <td>3376</td>\n",
       "      <td>183.7</td>\n",
       "      <td>11.80</td>\n",
       "      <td>0.000</td>\n",
       "      <td>6.04</td>\n",
       "      <td>ENSG00000268895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>659-693</td>\n",
       "      <td>517</td>\n",
       "      <td>3114</td>\n",
       "      <td>461.4</td>\n",
       "      <td>17.80</td>\n",
       "      <td>0.001</td>\n",
       "      <td>3.12</td>\n",
       "      <td>ENSG00000268895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>130-150</td>\n",
       "      <td>151</td>\n",
       "      <td>1999</td>\n",
       "      <td>117.6</td>\n",
       "      <td>9.64</td>\n",
       "      <td>0.001</td>\n",
       "      <td>3.47</td>\n",
       "      <td>ENSG00000268895_m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>130-150</td>\n",
       "      <td>108</td>\n",
       "      <td>1418</td>\n",
       "      <td>86.6</td>\n",
       "      <td>8.35</td>\n",
       "      <td>0.002</td>\n",
       "      <td>2.57</td>\n",
       "      <td>ENSG00000268895_p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>2583-2618</td>\n",
       "      <td>102</td>\n",
       "      <td>978</td>\n",
       "      <td>82.8</td>\n",
       "      <td>8.25</td>\n",
       "      <td>0.007</td>\n",
       "      <td>2.32</td>\n",
       "      <td>ENSG00000271270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>1367-1387</td>\n",
       "      <td>73</td>\n",
       "      <td>1007</td>\n",
       "      <td>54.6</td>\n",
       "      <td>7.06</td>\n",
       "      <td>0.009</td>\n",
       "      <td>2.60</td>\n",
       "      <td>ENSG00000271270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>1822-1869</td>\n",
       "      <td>127</td>\n",
       "      <td>953</td>\n",
       "      <td>105.6</td>\n",
       "      <td>9.34</td>\n",
       "      <td>0.015</td>\n",
       "      <td>2.29</td>\n",
       "      <td>ENSG00000271270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>1822-1869</td>\n",
       "      <td>84</td>\n",
       "      <td>585</td>\n",
       "      <td>65.8</td>\n",
       "      <td>7.62</td>\n",
       "      <td>0.009</td>\n",
       "      <td>2.39</td>\n",
       "      <td>ENSG00000271270_m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>2583-2618</td>\n",
       "      <td>66</td>\n",
       "      <td>603</td>\n",
       "      <td>52.2</td>\n",
       "      <td>6.56</td>\n",
       "      <td>0.018</td>\n",
       "      <td>2.11</td>\n",
       "      <td>ENSG00000271270_m</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>76 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          DBD  tr_with DBS  tr_without DBS  ntr_with DBS(average)  \\\n",
       "21  2406-2425           68             882                   45.1   \n",
       "20  1537-1558           73             877                   58.9   \n",
       "19  1980-2011          289             661                  250.2   \n",
       "47       9-28           79            3084                   63.2   \n",
       "18       9-28           51            1883                   40.9   \n",
       "64      84-96          724            5848                  496.3   \n",
       "65    108-129          108            6464                   85.2   \n",
       "67    240-252           68            6504                   43.0   \n",
       "68    339-356           75            6497                   56.0   \n",
       "69    733-749          155            6417                  113.0   \n",
       "66    178-195          136            6436                  104.9   \n",
       "71    920-937          146            6426                  102.9   \n",
       "73  1019-1040          190            6382                  140.7   \n",
       "72    955-972          248            6324                  192.3   \n",
       "70    901-918          210            6362                  155.4   \n",
       "42      84-96          425            3251                  327.6   \n",
       "41    955-972          142            3534                  124.6   \n",
       "43    920-937           90            3586                   66.3   \n",
       "39    733-749           94            3582                   73.4   \n",
       "44  1019-1040          114            3562                   89.3   \n",
       "40    108-129           67            3609                   53.7   \n",
       "3     901-918          123            3192                   89.5   \n",
       "2       84-96          337            2978                  310.8   \n",
       "46  2539-2558          223            1004                  200.7   \n",
       "45  2433-2446           59            1168                   44.0   \n",
       "5     681-693           86            1001                   69.0   \n",
       "6   1052-1067           80            1007                   62.4   \n",
       "4     116-144           73            1014                   55.9   \n",
       "8     681-693           55             607                   42.5   \n",
       "37  2513-2588          401             723                  363.2   \n",
       "..        ...          ...             ...                    ...   \n",
       "25      46-94         1186            6147                  827.2   \n",
       "54  1153-1174          237            3685                  189.9   \n",
       "53  1079-1100          568            3354                  509.5   \n",
       "51  1904-1923          178            3744                  160.4   \n",
       "52    521-541          276            3646                  237.6   \n",
       "49  1428-1444          162            3760                  135.0   \n",
       "48      46-94          619            3303                  549.1   \n",
       "50  1732-1744           90            3832                   77.1   \n",
       "12  1667-1680           60            3361                   48.2   \n",
       "13  1904-1923          159            3262                  141.8   \n",
       "14  1732-1744           81            3340                   68.6   \n",
       "15      46-94          564            2857                  482.1   \n",
       "16  1079-1100          524            2897                  447.5   \n",
       "17  1825-1846          161            3260                  117.0   \n",
       "9   1428-1444          146            3275                  119.4   \n",
       "11  1153-1174          196            3225                  172.1   \n",
       "10    521-541          244            3177                  212.4   \n",
       "61  1509-1535          130            1008                  104.4   \n",
       "62    440-457           97            1041                   77.9   \n",
       "63    694-713          162             976                  130.7   \n",
       "24  2259-2290          253             930                  231.3   \n",
       "23    130-150          255            3376                  183.7   \n",
       "22    659-693          517            3114                  461.4   \n",
       "7     130-150          151            1999                  117.6   \n",
       "1     130-150          108            1418                   86.6   \n",
       "55  2583-2618          102             978                   82.8   \n",
       "56  1367-1387           73            1007                   54.6   \n",
       "57  1822-1869          127             953                  105.6   \n",
       "74  1822-1869           84             585                   65.8   \n",
       "75  2583-2618           66             603                   52.2   \n",
       "\n",
       "    ntr_with DBS(std)  p-value  z-score                lnc  \n",
       "21               6.29    0.000     3.63    ENSG00000212978  \n",
       "20               7.25    0.024     1.94    ENSG00000212978  \n",
       "19              13.40    0.001     2.88    ENSG00000212978  \n",
       "47               7.09    0.012     2.24    ENSG00000214548  \n",
       "18               5.86    0.031     1.72  ENSG00000214548_p  \n",
       "64              15.80    0.000    14.40    ENSG00000223485  \n",
       "65               6.75    0.000     3.39    ENSG00000223485  \n",
       "67               4.81    0.000     5.20    ENSG00000223485  \n",
       "68               5.56    0.000     3.42    ENSG00000223485  \n",
       "69               7.79    0.000     5.40    ENSG00000223485  \n",
       "66               7.38    0.000     4.21    ENSG00000223485  \n",
       "71               7.22    0.000     5.98    ENSG00000223485  \n",
       "73               8.49    0.000     5.81    ENSG00000223485  \n",
       "72               9.78    0.000     5.70    ENSG00000223485  \n",
       "70               9.38    0.000     5.82    ENSG00000223485  \n",
       "42              15.00    0.000     6.49  ENSG00000223485_m  \n",
       "41               9.44    0.031     1.84  ENSG00000223485_m  \n",
       "43               7.30    0.000     3.25  ENSG00000223485_m  \n",
       "39               7.20    0.001     2.87  ENSG00000223485_m  \n",
       "44               8.19    0.000     3.02  ENSG00000223485_m  \n",
       "40               6.40    0.009     2.08  ENSG00000223485_m  \n",
       "3                8.32    0.000     4.03  ENSG00000223485_p  \n",
       "2               15.20    0.036     1.72  ENSG00000223485_p  \n",
       "46              12.60    0.034     1.77    ENSG00000225733  \n",
       "45               6.30    0.009     2.38    ENSG00000225733  \n",
       "5                7.79    0.009     2.19    ENSG00000229852  \n",
       "6                7.34    0.014     2.40    ENSG00000229852  \n",
       "4                6.81    0.002     2.50    ENSG00000229852  \n",
       "8                6.15    0.018     2.04  ENSG00000229852_m  \n",
       "37              14.90    0.005     2.54    ENSG00000230630  \n",
       "..                ...      ...      ...                ...  \n",
       "25              18.30    0.000    19.60    ENSG00000233396  \n",
       "54              12.10    0.000     3.91  ENSG00000233396_m  \n",
       "53              18.00    0.000     3.24  ENSG00000233396_m  \n",
       "51              10.60    0.044     1.65  ENSG00000233396_m  \n",
       "52              12.50    0.000     3.07  ENSG00000233396_m  \n",
       "49               9.71    0.002     2.78  ENSG00000233396_m  \n",
       "48              18.60    0.001     3.76  ENSG00000233396_m  \n",
       "50               7.63    0.043     1.70  ENSG00000233396_m  \n",
       "12               6.16    0.025     1.92  ENSG00000233396_p  \n",
       "13              10.20    0.035     1.68  ENSG00000233396_p  \n",
       "14               7.23    0.049     1.71  ENSG00000233396_p  \n",
       "15              17.90    0.000     4.57  ENSG00000233396_p  \n",
       "16              17.80    0.000     4.29  ENSG00000233396_p  \n",
       "17               9.39    0.000     4.69  ENSG00000233396_p  \n",
       "9                9.42    0.003     2.82  ENSG00000233396_p  \n",
       "11              11.50    0.020     2.09  ENSG00000233396_p  \n",
       "10              12.20    0.005     2.58  ENSG00000233396_p  \n",
       "61               9.57    0.003     2.67    ENSG00000240875  \n",
       "62               8.16    0.013     2.34    ENSG00000240875  \n",
       "63              10.30    0.000     3.04    ENSG00000240875  \n",
       "24              13.30    0.043     1.63    ENSG00000260032  \n",
       "23              11.80    0.000     6.04    ENSG00000268895  \n",
       "22              17.80    0.001     3.12    ENSG00000268895  \n",
       "7                9.64    0.001     3.47  ENSG00000268895_m  \n",
       "1                8.35    0.002     2.57  ENSG00000268895_p  \n",
       "55               8.25    0.007     2.32    ENSG00000271270  \n",
       "56               7.06    0.009     2.60    ENSG00000271270  \n",
       "57               9.34    0.015     2.29    ENSG00000271270  \n",
       "74               7.62    0.009     2.39  ENSG00000271270_m  \n",
       "75               6.56    0.018     2.11  ENSG00000271270_m  \n",
       "\n",
       "[76 rows x 8 columns]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "res.sort_values(['lnc'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ENSG00000212978',\n",
       " 'ENSG00000214548',\n",
       " 'ENSG00000214548_p',\n",
       " 'ENSG00000223485',\n",
       " 'ENSG00000223485_m',\n",
       " 'ENSG00000223485_p',\n",
       " 'ENSG00000225733',\n",
       " 'ENSG00000229852',\n",
       " 'ENSG00000229852_m',\n",
       " 'ENSG00000230630',\n",
       " 'ENSG00000230630_m',\n",
       " 'ENSG00000233117',\n",
       " 'ENSG00000233117_p',\n",
       " 'ENSG00000233396',\n",
       " 'ENSG00000233396_m',\n",
       " 'ENSG00000233396_p',\n",
       " 'ENSG00000240875',\n",
       " 'ENSG00000260032',\n",
       " 'ENSG00000268895',\n",
       " 'ENSG00000268895_m',\n",
       " 'ENSG00000268895_p',\n",
       " 'ENSG00000271270',\n",
       " 'ENSG00000271270_m']"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sorted(list(res['lnc'].unique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "fantom_lnc_tdf = ['ENSG00000212978', 'ENSG00000214548', 'ENSG00000223485', 'ENSG00000225733', \n",
    "                 'ENSG00000229852', 'ENSG00000230630', 'ENSG00000233117', 'ENSG00000233396',\n",
    "                 'ENSG00000240875', 'ENSG00000260032', 'ENSG00000268895', 'ENSG00000271270']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>lncRNAId</th>\n",
       "      <th>lncRNAName</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>mm_pvalue</th>\n",
       "      <th>mp_pvalue</th>\n",
       "      <th>pm_pvalue</th>\n",
       "      <th>pp_pvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000233117</td>\n",
       "      <td>LINC00702</td>\n",
       "      <td>4.692277e-08</td>\n",
       "      <td>8.878292e-28</td>\n",
       "      <td>5.876098e-07</td>\n",
       "      <td>6.352252e-06</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000212978</td>\n",
       "      <td>AC016747.3</td>\n",
       "      <td>7.310864e-05</td>\n",
       "      <td>4.326550e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>8.976846e-14</td>\n",
       "      <td>0.361277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000260032</td>\n",
       "      <td>LINC00657</td>\n",
       "      <td>4.423346e-05</td>\n",
       "      <td>4.600410e-01</td>\n",
       "      <td>6.213641e-06</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.173855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000223485</td>\n",
       "      <td>RP11-417E7.1</td>\n",
       "      <td>1.468296e-11</td>\n",
       "      <td>2.535745e-31</td>\n",
       "      <td>4.455333e-20</td>\n",
       "      <td>1.157244e-37</td>\n",
       "      <td>0.000005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000271270</td>\n",
       "      <td>TMCC1-AS1</td>\n",
       "      <td>4.082303e-03</td>\n",
       "      <td>2.142700e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>2.122146e-05</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENSG00000268895</td>\n",
       "      <td>A1BG-AS1</td>\n",
       "      <td>4.290017e-11</td>\n",
       "      <td>1.476438e-22</td>\n",
       "      <td>1.866638e-07</td>\n",
       "      <td>3.154033e-20</td>\n",
       "      <td>0.238645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ENSG00000225733</td>\n",
       "      <td>FGD5-AS1</td>\n",
       "      <td>3.378603e-09</td>\n",
       "      <td>2.163507e-03</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>4.037441e-16</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENSG00000240875</td>\n",
       "      <td>LINC00886</td>\n",
       "      <td>6.833321e-03</td>\n",
       "      <td>2.163507e-03</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>9.296805e-01</td>\n",
       "      <td>0.378362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ENSG00000230630</td>\n",
       "      <td>DNM3OS</td>\n",
       "      <td>1.006800e-15</td>\n",
       "      <td>6.105785e-36</td>\n",
       "      <td>3.586202e-05</td>\n",
       "      <td>2.390112e-03</td>\n",
       "      <td>0.085100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ENSG00000214548</td>\n",
       "      <td>MEG3</td>\n",
       "      <td>1.203857e-12</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>7.552614e-13</td>\n",
       "      <td>1.639964e-02</td>\n",
       "      <td>0.295756</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ENSG00000233396</td>\n",
       "      <td>RP11-458D21.1</td>\n",
       "      <td>6.878953e-08</td>\n",
       "      <td>7.219693e-02</td>\n",
       "      <td>9.904556e-20</td>\n",
       "      <td>1.816320e-05</td>\n",
       "      <td>0.010025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ENSG00000229852</td>\n",
       "      <td>RP11-398K22.12</td>\n",
       "      <td>4.668782e-03</td>\n",
       "      <td>1.606057e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.163267e-05</td>\n",
       "      <td>0.140823</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           lncRNAId      lncRNAName        pvalue     mm_pvalue     mp_pvalue  \\\n",
       "0   ENSG00000233117       LINC00702  4.692277e-08  8.878292e-28  5.876098e-07   \n",
       "1   ENSG00000212978      AC016747.3  7.310864e-05  4.326550e-01  1.000000e+00   \n",
       "2   ENSG00000260032       LINC00657  4.423346e-05  4.600410e-01  6.213641e-06   \n",
       "3   ENSG00000223485    RP11-417E7.1  1.468296e-11  2.535745e-31  4.455333e-20   \n",
       "4   ENSG00000271270       TMCC1-AS1  4.082303e-03  2.142700e-01  1.000000e+00   \n",
       "6   ENSG00000268895        A1BG-AS1  4.290017e-11  1.476438e-22  1.866638e-07   \n",
       "8   ENSG00000225733        FGD5-AS1  3.378603e-09  2.163507e-03  1.000000e+00   \n",
       "9   ENSG00000240875       LINC00886  6.833321e-03  2.163507e-03  1.000000e+00   \n",
       "13  ENSG00000230630          DNM3OS  1.006800e-15  6.105785e-36  3.586202e-05   \n",
       "15  ENSG00000214548            MEG3  1.203857e-12  1.000000e+00  7.552614e-13   \n",
       "16  ENSG00000233396   RP11-458D21.1  6.878953e-08  7.219693e-02  9.904556e-20   \n",
       "18  ENSG00000229852  RP11-398K22.12  4.668782e-03  1.606057e-01  1.000000e+00   \n",
       "\n",
       "       pm_pvalue  pp_pvalue  \n",
       "0   6.352252e-06   1.000000  \n",
       "1   8.976846e-14   0.361277  \n",
       "2   1.000000e+00   0.173855  \n",
       "3   1.157244e-37   0.000005  \n",
       "4   2.122146e-05   1.000000  \n",
       "6   3.154033e-20   0.238645  \n",
       "8   4.037441e-16   1.000000  \n",
       "9   9.296805e-01   0.378362  \n",
       "13  2.390112e-03   0.085100  \n",
       "15  1.639964e-02   0.295756  \n",
       "16  1.816320e-05   0.010025  \n",
       "18  1.163267e-05   0.140823  "
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rnas[rnas['lncRNAId'].isin(fantom_lnc_tdf)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#1)Посчитать отдельно с двумя буквами варианты, найти подходящий ccf\n",
    "#2)Статистически проверить что отношение таргетов к общему становится уже с увеличением фильтрации\n",
    "#3)Подбирать ccf динамически(пункт 1), возможно нужны параметры G и C для триплексатора"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
