{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import h5py\n",
    "import scipy.stats as stats\n",
    "import numpy as np\n",
    "from collections import defaultdict\n",
    "from statsmodels.stats.multitest import multipletests\n",
    "\n",
    "from BCBio import GFF\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqUtils import GC\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.SeqFeature import SeqFeature, FeatureLocation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "targets = [(\"H3K27ac\", \"_narrow\"), (\"H3K27me3\", \"\"), (\"H3K36me3\", \"\"), \n",
    "           (\"H3K4me1\", \"_narrow\"), (\"H3K4me2\", \"_narrow\"), (\"H3K4me3\", \"_narrow\"), (\"H3K79me2\", \"\"), \n",
    "           (\"H3K9ac\", \"_narrow\"), (\"H3K9me3\", \"\"), (\"H4K20me1\", \"\")]\n",
    "           #, (\"methylation\", \"\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getLncRNAGeneSet():\n",
    "    \n",
    "    in_file = \"../all_marks/gencode.v31.long_noncoding_RNAs.gff3\"\n",
    "    in_handle = open(in_file)\n",
    "\n",
    "    limit_info = dict(\n",
    "        gff_id = ['chr1', 'chr10', 'chr11', 'chr12', 'chr13', 'chr14', 'chr15', 'chr16', 'chr17', 'chr18', \n",
    "     'chr19', 'chr2', 'chr20', 'chr21', 'chr22', 'chr3', 'chr4', 'chr5', 'chr6', 'chr7', 'chr8', 'chr9', 'chrX', 'chrY'],\n",
    "        gff_type = [\"gene\"])\n",
    "\n",
    "    lncRNAgenes = []\n",
    "    for rec in GFF.parse(in_handle, limit_info=limit_info):\n",
    "        lncRNAgenes.append((rec.id, rec.features)) \n",
    "\n",
    "    in_handle.close()\n",
    "    \n",
    "    lncRNAgenesStartPositions = {\n",
    "    g.id.split(\".\")[0]:[i, g.location.start.position, g.location.end.position, g.strand] for i, j in lncRNAgenes for g in j}\n",
    "    \n",
    "    return [i for i in lncRNAgenesStartPositions.keys() if i.startswith(\"ENSG\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Генерим бэкграунд для конкретного эксперимента: пересекаем наш список lncRNA(из которого мы выбирали)\n",
    "#с экспериментальным списком генов, которые в принципе экспрессировались\n",
    "#Если передаем несколько бэкграундов, значит хотим их объеденить\n",
    "def getCommonBackground(*exp_backgrounds):\n",
    "    lncRNASet = set(getLncRNAGeneSet())\n",
    "    exp_background = None\n",
    "    \n",
    "    for exp in exp_backgrounds:\n",
    "        lineList = [line.rstrip('\\n') for line in open(exp)]\n",
    "        lineList = [g for g in lineList if g.startswith(\"ENSG\")]\n",
    "        if exp_background is not None:\n",
    "            exp_background = exp_background.union(set([r for l in [g.split(',') for g in lineList] for r in l]))\n",
    "        else:\n",
    "            exp_background = set([r for l in [g.split(',') for g in lineList] for r in l])\n",
    "    \n",
    "    return lncRNASet.intersection(exp_background)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getCommonGencodeBackground(target, cell):\n",
    "    \"\"\"\n",
    "    В данном случае у эксперимента бэкграунд - вся gencode v31 аннотация, а у наших корреляций бэкграунд у каждой\n",
    "    метки свой - кол-во lncRNA, у которых искали корреляциий. Т.к. этот список тоже из gencode v31, то\n",
    "    в качестве бэкграунда у нас просто будут lncRNA из соответствующего метке lncRNA_matrix_filtered_norm.csv\n",
    "    файла.\n",
    "    \n",
    "    + Решили фильтровать гены по экспресии в клетке: берем только те, что экспрессируются\n",
    "    \"\"\"\n",
    "\n",
    "    l = pd.read_csv(\"/data/mazurovev/all_marks/\" + target + \"/lncRNA_matrix_filtered_norm.csv\", sep=\"\\t\", index_col=0)\n",
    "    cell_exp = set()\n",
    "    for c in cell:\n",
    "        cell_exp = cell_exp.union(set([g.split(\".\")[0] for g in pd.read_csv(\"../eCLIP/\" + c + \".eCLIP.detectable.genes.tsv\", header=None)[0].tolist()]))\n",
    "    \n",
    "    # print(\"background size = \" + str(len(set(l.index).intersection(cell_exp))))\n",
    "    \n",
    "    return set(l.index).intersection(cell_exp)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 322,
   "metadata": {},
   "outputs": [],
   "source": [
    "t = pd.read_csv(\"/data/mazurovev/all_marks/\" + \"H3K27ac\" + \"/lncRNA_matrix_filtered_norm.csv\", sep=\"\\t\", index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 323,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3094"
      ]
     },
     "execution_count": 323,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(t.index).intersection(set(s)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 319,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(22471, 1)"
      ]
     },
     "execution_count": 319,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"../eCLIP/\" + \"K562\" + \".eCLIP.detectable.genes.tsv\", header=None).shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 320,
   "metadata": {},
   "outputs": [],
   "source": [
    "s = [g.split(\".\")[0] for g in pd.read_csv(\"../eCLIP/\" + \"K562\" + \".c.genes.tsv\", header=None)[0].tolist()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 321,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "22471"
      ]
     },
     "execution_count": 321,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(s)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Возвращаем список мишеней для конкретного белка(в конкретной клетке), которые так же есть в бэкграунде\n",
    "#Если белков несколько(могут быть из разных клеток), то сливаем их в один список\n",
    "def getTargetsList(*protein_files):\n",
    "    targets_list = None\n",
    "    \n",
    "    for p in protein_files:\n",
    "        print(\"Get targets list for \" + p)\n",
    "        pretargets_list = pd.read_csv(\"../eCLIP/\" + p, sep=\"\\t\")\n",
    "        #pretargets_list.head()\n",
    "        pretargets_list = [r.split(',') for r in pretargets_list[pretargets_list['Gene_ID'].str.startswith(\"ENSG\")]['Gene_ID'].tolist()]\n",
    "        #print(pretargets_list)\n",
    "        if targets_list is not None:\n",
    "            targets_list = targets_list.union(set([r.split(\".\")[0] for l in pretargets_list for r in l]))\n",
    "        else:\n",
    "            targets_list = set([r.split(\".\")[0] for l in pretargets_list for r in l])\n",
    "     \n",
    "    #res = targets_list.intersection(background)\n",
    "    \n",
    "    print(\"Targets list len \" + str(len(targets_list)))\n",
    "    return targets_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getCorrsLncRNAsLists(target):\n",
    "    def getCorrsLncRNAsLists(lncRNAnames, corrs):\n",
    "        plus_corrs_idx = list(np.where(np.sum(corrs > 0, axis=1) == 0)[0])\n",
    "        plus_hm_lncRNAs = lncRNAnames[plus_corrs_idx]\n",
    "        minus_corrs_idx = list(np.where(np.sum(corrs < 0, axis=1) == 0)[0])\n",
    "        minus_hm_lncRNAs = lncRNAnames[minus_corrs_idx]\n",
    "        #mixture_idx = [i for i in range(data.shape[0]) if i not in (plus_corrs_idx + minus_corrs_idx)]\n",
    "        #mixture_hm_lncRNAs = set(lncRNAnames[mixture_idx]).intersection(background)\n",
    "   \n",
    "        return {\"general\": lncRNAnames, \"only + corrs\": plus_hm_lncRNAs, \"only - corrs\": minus_hm_lncRNAs}\n",
    "    \n",
    "\n",
    "    print(\"makes count for \" + target)\n",
    "    if(target == \"methylation\"):\n",
    "        corrs_lncRNAs = defaultdict(list)\n",
    "        for i in range(0, 19):\n",
    "            with h5py.File(\"/data/mazurovev/all_marks/\" + target + \"/lncRNA_Peaks_corrs/lncRNA_Peaks_Correlations_corrected_non_zero_\" + str(i) + \".hdf5\", 'r') as f:\n",
    "                tmp = getCorrsLncRNAsLists(f['lncRNAs_names'][:], f['corrs_matrix'][:])\n",
    "                corrs_lncRNAs[\"general\"].append(tmp[\"general\"])\n",
    "                corrs_lncRNAs[\"only + corrs\"].append(tmp[\"only + corrs\"])\n",
    "                corrs_lncRNAs[\"only - corrs\"].append(tmp[\"only - corrs\"])\n",
    "    else:\n",
    "        with h5py.File(\"/data/mazurovev/all_marks/\" + target + \"/lncRNA_Peaks_corrs/lncRNA_Peaks_Correlations_corrected_non_zero.hdf5\", 'r') as f:\n",
    "            corrs_lncRNAs = getCorrsLncRNAsLists(f['lncRNAs_names'][:], f['corrs_matrix'][:])\n",
    "            \n",
    "    return corrs_lncRNAs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "makes count for H3K27ac\n",
      "makes count for H3K27me3\n",
      "makes count for H3K36me3\n",
      "makes count for H3K4me1\n",
      "makes count for H3K4me2\n",
      "makes count for H3K4me3\n",
      "makes count for H3K79me2\n",
      "makes count for H3K9ac\n",
      "makes count for H3K9me3\n",
      "makes count for H4K20me1\n"
     ]
    }
   ],
   "source": [
    "corrs_lncRNAs = {}\n",
    "for target in targets:\n",
    "    corrs_lncRNAs[target[0]] = getCorrsLncRNAsLists(target[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "def makeCounts(eclip, eclip_targets, cell):\n",
    "\n",
    "    res = []\n",
    "    for target in targets:\n",
    "        background = getCommonGencodeBackground(target[0], cell)\n",
    "        for list_type in  corrs_lncRNAs[target[0]].keys():\n",
    "            type_list = set(corrs_lncRNAs[target[0]][list_type]).intersection(background)\n",
    "            res.append((target[0], list_type, \n",
    "                        len(type_list), \n",
    "                        eclip, len(eclip_targets.intersection(background)), \n",
    "                        len(type_list.intersection(set(eclip_targets))),\n",
    "                       len(background)))\n",
    "            \n",
    "                \n",
    "    df = pd.DataFrame(res, columns =['Modification', 'Modification lncRNAs list type', 'Modification lncRNAs list count', 'RBP', 'RBP lncRNAs count', 'TP', 'background'])\n",
    "    df['TN'] = df['background'] - df['Modification lncRNAs list count'] - df['RBP lncRNAs count'] + df['TP']\n",
    "    df['FP'] = df['Modification lncRNAs list count'] - df['TP']\n",
    "    df['FN'] = df['RBP lncRNAs count'] - df['TP']\n",
    "    \n",
    "    # Уберем строки, где TP == 0 или какой-нибудь из каунтов == 0\n",
    "    df = df[(df['TP'] != 0) & (df[\"Modification lncRNAs list count\"] != 0) & (df[\"RBP lncRNAs count\"] != 0)]\n",
    "    \n",
    "    return res, df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "def count_p_values(df):\n",
    "    pv_list = []\n",
    "    oddsratio_list = []\n",
    "    for index, row in df.iterrows():\n",
    "        oddsratio, pv = stats.fisher_exact([[row['TP'], row['FP']], [row['FN'], row['TN']]], alternative='greater')\n",
    "        pv_list.append(pv)\n",
    "        oddsratio_list.append(oddsratio)\n",
    "    \n",
    "    df['p-value'] = pv_list\n",
    "    df['oddsratio'] = oddsratio_list\n",
    "    correct = multipletests(pv_list, alpha=0.05, method='fdr_bh')\n",
    "    df['Correction'] = correct[0]\n",
    "    \n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getResults(proteins, proteins_names, cell):\n",
    "    res = []\n",
    "\n",
    "    for protein, name in zip(proteins, proteins_names):\n",
    "        if(isinstance(protein, list)):\n",
    "            targets = getTargetsList(*protein)\n",
    "        else:\n",
    "            targets = getTargetsList(protein)\n",
    "\n",
    "        _, counts = makeCounts(name, targets, cell)\n",
    "        if counts.shape[0] != 0:\n",
    "            res.append((name, count_p_values(counts)))\n",
    "    \n",
    "    concat = []\n",
    "    without_corr = []\n",
    "    for name, df in res:\n",
    "        # display(df)\n",
    "        concat.append(df[df['Correction'] == True])\n",
    "        without_corr.append(df[df['p-value'] < 0.05])\n",
    "        \n",
    "    return res, pd.concat(concat), pd.concat(without_corr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 337,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Get targets list for KHDRBS1new_K562.txt\n",
      "Targets list len 881\n",
      "Get targets list for HNRNPLnew_K562.txt\n",
      "Targets list len 1837\n",
      "Get targets list for TAF15new_K562.txt\n",
      "Targets list len 791\n",
      "Get targets list for HNRNPUnew_K562.txt\n",
      "Targets list len 1122\n",
      "Get targets list for HNRNPUL1new_K562.txt\n",
      "Targets list len 855\n",
      "Get targets list for SAFB2new_K562.txt\n",
      "Targets list len 2577\n",
      "Get targets list for NONOnew_K562.txt\n",
      "Targets list len 2448\n",
      "Get targets list for SAFBnew_K562.txt\n",
      "Targets list len 1779\n",
      "Get targets list for HNRNPMnew_K562.txt\n",
      "Targets list len 171\n"
     ]
    }
   ],
   "source": [
    "proteins = [f for f in os.listdir(\"../eCLIP/\") if \"txt\" in f and \"K562\" in f and \"new\" in f]\n",
    "proteins_names = [os.path.splitext(p)[0] for p in proteins]\n",
    "\n",
    "res_k562, k562, p_k562 = getResults(proteins, proteins_names, [\"K562\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 340,
   "metadata": {},
   "outputs": [],
   "source": [
    "k562.to_csv(\"../eCLIP/K562_norm_bg_26.10.2020.tsv\", sep=\"\\t\", index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 341,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Get targets list for HNRNPUL1new_HepG2.txt\n",
      "Targets list len 77\n",
      "Get targets list for ILF3new_HepG2.txt\n",
      "Targets list len 717\n",
      "Get targets list for HNRNPLnew_HepG2.txt\n",
      "Targets list len 814\n",
      "Get targets list for QKInew_HepG2.txt\n",
      "Targets list len 492\n",
      "Get targets list for SLTMnew_HepG2.txt\n",
      "Targets list len 777\n",
      "Get targets list for NCBP2new_HepG2.txt\n",
      "Targets list len 97\n",
      "Get targets list for CSTF2Tnew_HepG2.txt\n",
      "Targets list len 2597\n",
      "Get targets list for SUGP2new_HepG2.txt\n",
      "Targets list len 2264\n",
      "Get targets list for MATR3new_HepG2.txt\n",
      "Targets list len 1094\n",
      "Get targets list for CSTF2new_HepG2.txt\n",
      "Targets list len 494\n",
      "Get targets list for DROSHAnew_HepG2.txt\n",
      "Targets list len 94\n",
      "Get targets list for NKRFnew_HepG2.txt\n",
      "Targets list len 260\n",
      "Get targets list for SFPQnew_HepG2.txt\n",
      "Targets list len 1058\n",
      "Get targets list for FUSnew_HepG2.txt\n",
      "Targets list len 70\n",
      "Get targets list for HNRNPUnew_HepG2.txt\n",
      "Targets list len 1197\n",
      "Get targets list for FAM120Anew_HepG2.txt\n",
      "Targets list len 72\n",
      "Get targets list for HNRNPCnew_HepG2.txt\n",
      "Targets list len 767\n",
      "Get targets list for SAFBnew_HepG2.txt\n",
      "Targets list len 1446\n",
      "Get targets list for TAF15new_HepG2.txt\n",
      "Targets list len 167\n",
      "Get targets list for HNRNPA1new_HepG2.txt\n",
      "Targets list len 673\n"
     ]
    }
   ],
   "source": [
    "proteins = [f for f in os.listdir(\"../eCLIP/\") if \"txt\" in f and \"HepG2\" in f and \"new\" in f]\n",
    "proteins_names = [os.path.splitext(p)[0] for p in proteins]\n",
    "\n",
    "res_hepG2, hepG2, p = getResults(proteins, proteins_names, [\"HepG2\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 346,
   "metadata": {},
   "outputs": [],
   "source": [
    "hepG2.to_csv(\"../eCLIP/HepG2_norm_bg_26.10.2020.tsv\", sep=\"\\t\", index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 351,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Get targets list for HNRNPLnew_K562.txt\n",
      "Get targets list for HNRNPLnew_HepG2.txt\n",
      "Targets list len 2146\n",
      "Get targets list for HNRNPUnew_K562.txt\n",
      "Get targets list for HNRNPUnew_HepG2.txt\n",
      "Targets list len 1799\n",
      "Get targets list for HNRNPUL1new_HepG2.txt\n",
      "Get targets list for HNRNPUL1new_K562.txt\n",
      "Targets list len 885\n",
      "Get targets list for SAFBnew_K562.txt\n",
      "Get targets list for SAFBnew_HepG2.txt\n",
      "Targets list len 2497\n",
      "Get targets list for TAF15new_K562.txt\n",
      "Get targets list for TAF15new_HepG2.txt\n",
      "Targets list len 882\n"
     ]
    }
   ],
   "source": [
    "proteins_names = ['HNRNPLnew', 'HNRNPUnew', 'HNRNPUL1new', 'SAFBnew', 'TAF15new']\n",
    "proteins = []\n",
    "\n",
    "for name in proteins_names:\n",
    "    proteins.append([f for f in os.listdir(\"../eCLIP/\") if \"txt\" in f and name == os.path.splitext(f)[0].split('_')[0]])\n",
    "    \n",
    "res_union, union, p = getResults(proteins, proteins_names, [\"K562\", \"HepG2\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 352,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Modification</th>\n",
       "      <th>Modification lncRNAs list type</th>\n",
       "      <th>Modification lncRNAs list count</th>\n",
       "      <th>RBP</th>\n",
       "      <th>RBP lncRNAs count</th>\n",
       "      <th>TP</th>\n",
       "      <th>background</th>\n",
       "      <th>TN</th>\n",
       "      <th>FP</th>\n",
       "      <th>FN</th>\n",
       "      <th>p-value</th>\n",
       "      <th>oddsratio</th>\n",
       "      <th>Correction</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>H3K27ac</td>\n",
       "      <td>general</td>\n",
       "      <td>1335</td>\n",
       "      <td>HNRNPLnew</td>\n",
       "      <td>413</td>\n",
       "      <td>172</td>\n",
       "      <td>3769</td>\n",
       "      <td>2193</td>\n",
       "      <td>1163</td>\n",
       "      <td>241</td>\n",
       "      <td>3.209960e-03</td>\n",
       "      <td>1.345768</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>H3K4me1</td>\n",
       "      <td>general</td>\n",
       "      <td>1290</td>\n",
       "      <td>HNRNPLnew</td>\n",
       "      <td>418</td>\n",
       "      <td>166</td>\n",
       "      <td>3840</td>\n",
       "      <td>2298</td>\n",
       "      <td>1124</td>\n",
       "      <td>252</td>\n",
       "      <td>3.224187e-03</td>\n",
       "      <td>1.346763</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>H3K27ac</td>\n",
       "      <td>general</td>\n",
       "      <td>1335</td>\n",
       "      <td>HNRNPUnew</td>\n",
       "      <td>559</td>\n",
       "      <td>253</td>\n",
       "      <td>3769</td>\n",
       "      <td>2128</td>\n",
       "      <td>1082</td>\n",
       "      <td>306</td>\n",
       "      <td>1.273295e-07</td>\n",
       "      <td>1.626086</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>H3K27me3</td>\n",
       "      <td>general</td>\n",
       "      <td>967</td>\n",
       "      <td>HNRNPUnew</td>\n",
       "      <td>562</td>\n",
       "      <td>172</td>\n",
       "      <td>3837</td>\n",
       "      <td>2480</td>\n",
       "      <td>795</td>\n",
       "      <td>390</td>\n",
       "      <td>9.935819e-04</td>\n",
       "      <td>1.375778</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>H3K4me1</td>\n",
       "      <td>general</td>\n",
       "      <td>1290</td>\n",
       "      <td>HNRNPUnew</td>\n",
       "      <td>563</td>\n",
       "      <td>221</td>\n",
       "      <td>3840</td>\n",
       "      <td>2208</td>\n",
       "      <td>1069</td>\n",
       "      <td>342</td>\n",
       "      <td>1.337199e-03</td>\n",
       "      <td>1.334712</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>H3K4me3</td>\n",
       "      <td>general</td>\n",
       "      <td>1168</td>\n",
       "      <td>HNRNPUnew</td>\n",
       "      <td>572</td>\n",
       "      <td>201</td>\n",
       "      <td>3929</td>\n",
       "      <td>2390</td>\n",
       "      <td>967</td>\n",
       "      <td>371</td>\n",
       "      <td>1.440612e-03</td>\n",
       "      <td>1.339040</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>H3K4me3</td>\n",
       "      <td>only - corrs</td>\n",
       "      <td>268</td>\n",
       "      <td>HNRNPUnew</td>\n",
       "      <td>572</td>\n",
       "      <td>55</td>\n",
       "      <td>3929</td>\n",
       "      <td>3144</td>\n",
       "      <td>213</td>\n",
       "      <td>517</td>\n",
       "      <td>3.740726e-03</td>\n",
       "      <td>1.570273</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>H3K27ac</td>\n",
       "      <td>general</td>\n",
       "      <td>1335</td>\n",
       "      <td>HNRNPUL1new</td>\n",
       "      <td>186</td>\n",
       "      <td>91</td>\n",
       "      <td>3769</td>\n",
       "      <td>2339</td>\n",
       "      <td>1244</td>\n",
       "      <td>95</td>\n",
       "      <td>7.233552e-05</td>\n",
       "      <td>1.801058</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>H3K27me3</td>\n",
       "      <td>general</td>\n",
       "      <td>967</td>\n",
       "      <td>HNRNPUL1new</td>\n",
       "      <td>187</td>\n",
       "      <td>73</td>\n",
       "      <td>3837</td>\n",
       "      <td>2756</td>\n",
       "      <td>894</td>\n",
       "      <td>114</td>\n",
       "      <td>1.327233e-05</td>\n",
       "      <td>1.974057</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>H3K36me3</td>\n",
       "      <td>general</td>\n",
       "      <td>879</td>\n",
       "      <td>HNRNPUL1new</td>\n",
       "      <td>187</td>\n",
       "      <td>60</td>\n",
       "      <td>3824</td>\n",
       "      <td>2818</td>\n",
       "      <td>819</td>\n",
       "      <td>127</td>\n",
       "      <td>2.177470e-03</td>\n",
       "      <td>1.625566</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>H3K4me1</td>\n",
       "      <td>general</td>\n",
       "      <td>1290</td>\n",
       "      <td>HNRNPUL1new</td>\n",
       "      <td>187</td>\n",
       "      <td>87</td>\n",
       "      <td>3840</td>\n",
       "      <td>2450</td>\n",
       "      <td>1203</td>\n",
       "      <td>100</td>\n",
       "      <td>1.163485e-04</td>\n",
       "      <td>1.771820</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>H3K4me3</td>\n",
       "      <td>general</td>\n",
       "      <td>1168</td>\n",
       "      <td>HNRNPUL1new</td>\n",
       "      <td>189</td>\n",
       "      <td>77</td>\n",
       "      <td>3929</td>\n",
       "      <td>2649</td>\n",
       "      <td>1091</td>\n",
       "      <td>112</td>\n",
       "      <td>6.114750e-04</td>\n",
       "      <td>1.669283</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>H3K9me3</td>\n",
       "      <td>general</td>\n",
       "      <td>743</td>\n",
       "      <td>HNRNPUL1new</td>\n",
       "      <td>186</td>\n",
       "      <td>56</td>\n",
       "      <td>3785</td>\n",
       "      <td>2912</td>\n",
       "      <td>687</td>\n",
       "      <td>130</td>\n",
       "      <td>3.013746e-04</td>\n",
       "      <td>1.825910</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>H3K27ac</td>\n",
       "      <td>general</td>\n",
       "      <td>1335</td>\n",
       "      <td>SAFBnew</td>\n",
       "      <td>507</td>\n",
       "      <td>238</td>\n",
       "      <td>3769</td>\n",
       "      <td>2165</td>\n",
       "      <td>1097</td>\n",
       "      <td>269</td>\n",
       "      <td>6.225594e-09</td>\n",
       "      <td>1.746127</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>H3K27me3</td>\n",
       "      <td>general</td>\n",
       "      <td>967</td>\n",
       "      <td>SAFBnew</td>\n",
       "      <td>512</td>\n",
       "      <td>163</td>\n",
       "      <td>3837</td>\n",
       "      <td>2521</td>\n",
       "      <td>804</td>\n",
       "      <td>349</td>\n",
       "      <td>1.666777e-04</td>\n",
       "      <td>1.464465</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>H3K4me1</td>\n",
       "      <td>general</td>\n",
       "      <td>1290</td>\n",
       "      <td>SAFBnew</td>\n",
       "      <td>511</td>\n",
       "      <td>215</td>\n",
       "      <td>3840</td>\n",
       "      <td>2254</td>\n",
       "      <td>1075</td>\n",
       "      <td>296</td>\n",
       "      <td>1.068543e-05</td>\n",
       "      <td>1.522973</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>H3K4me1</td>\n",
       "      <td>only - corrs</td>\n",
       "      <td>188</td>\n",
       "      <td>SAFBnew</td>\n",
       "      <td>511</td>\n",
       "      <td>38</td>\n",
       "      <td>3840</td>\n",
       "      <td>3179</td>\n",
       "      <td>150</td>\n",
       "      <td>473</td>\n",
       "      <td>4.407285e-03</td>\n",
       "      <td>1.702636</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>H3K4me3</td>\n",
       "      <td>general</td>\n",
       "      <td>1168</td>\n",
       "      <td>SAFBnew</td>\n",
       "      <td>518</td>\n",
       "      <td>191</td>\n",
       "      <td>3929</td>\n",
       "      <td>2434</td>\n",
       "      <td>977</td>\n",
       "      <td>327</td>\n",
       "      <td>1.042747e-04</td>\n",
       "      <td>1.455163</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>H3K4me3</td>\n",
       "      <td>only - corrs</td>\n",
       "      <td>268</td>\n",
       "      <td>SAFBnew</td>\n",
       "      <td>518</td>\n",
       "      <td>49</td>\n",
       "      <td>3929</td>\n",
       "      <td>3192</td>\n",
       "      <td>219</td>\n",
       "      <td>469</td>\n",
       "      <td>8.711866e-03</td>\n",
       "      <td>1.522797</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>H3K9me3</td>\n",
       "      <td>general</td>\n",
       "      <td>743</td>\n",
       "      <td>SAFBnew</td>\n",
       "      <td>509</td>\n",
       "      <td>125</td>\n",
       "      <td>3785</td>\n",
       "      <td>2658</td>\n",
       "      <td>618</td>\n",
       "      <td>384</td>\n",
       "      <td>1.937878e-03</td>\n",
       "      <td>1.400056</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>H3K27ac</td>\n",
       "      <td>general</td>\n",
       "      <td>1335</td>\n",
       "      <td>TAF15new</td>\n",
       "      <td>262</td>\n",
       "      <td>128</td>\n",
       "      <td>3769</td>\n",
       "      <td>2300</td>\n",
       "      <td>1207</td>\n",
       "      <td>134</td>\n",
       "      <td>2.536469e-06</td>\n",
       "      <td>1.820228</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>H3K27me3</td>\n",
       "      <td>general</td>\n",
       "      <td>967</td>\n",
       "      <td>TAF15new</td>\n",
       "      <td>263</td>\n",
       "      <td>95</td>\n",
       "      <td>3837</td>\n",
       "      <td>2702</td>\n",
       "      <td>872</td>\n",
       "      <td>168</td>\n",
       "      <td>2.974891e-05</td>\n",
       "      <td>1.752198</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>H3K4me1</td>\n",
       "      <td>general</td>\n",
       "      <td>1290</td>\n",
       "      <td>TAF15new</td>\n",
       "      <td>266</td>\n",
       "      <td>114</td>\n",
       "      <td>3840</td>\n",
       "      <td>2398</td>\n",
       "      <td>1176</td>\n",
       "      <td>152</td>\n",
       "      <td>6.904547e-04</td>\n",
       "      <td>1.529337</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>H3K4me2</td>\n",
       "      <td>general</td>\n",
       "      <td>823</td>\n",
       "      <td>TAF15new</td>\n",
       "      <td>215</td>\n",
       "      <td>87</td>\n",
       "      <td>2517</td>\n",
       "      <td>1566</td>\n",
       "      <td>736</td>\n",
       "      <td>128</td>\n",
       "      <td>7.519815e-03</td>\n",
       "      <td>1.446183</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>H3K4me3</td>\n",
       "      <td>general</td>\n",
       "      <td>1168</td>\n",
       "      <td>TAF15new</td>\n",
       "      <td>267</td>\n",
       "      <td>99</td>\n",
       "      <td>3929</td>\n",
       "      <td>2593</td>\n",
       "      <td>1069</td>\n",
       "      <td>168</td>\n",
       "      <td>4.518960e-03</td>\n",
       "      <td>1.429390</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Modification Modification lncRNAs list type  \\\n",
       "0       H3K27ac                        general   \n",
       "9       H3K4me1                        general   \n",
       "0       H3K27ac                        general   \n",
       "3      H3K27me3                        general   \n",
       "9       H3K4me1                        general   \n",
       "15      H3K4me3                        general   \n",
       "17      H3K4me3                   only - corrs   \n",
       "0       H3K27ac                        general   \n",
       "3      H3K27me3                        general   \n",
       "6      H3K36me3                        general   \n",
       "9       H3K4me1                        general   \n",
       "15      H3K4me3                        general   \n",
       "24      H3K9me3                        general   \n",
       "0       H3K27ac                        general   \n",
       "3      H3K27me3                        general   \n",
       "9       H3K4me1                        general   \n",
       "11      H3K4me1                   only - corrs   \n",
       "15      H3K4me3                        general   \n",
       "17      H3K4me3                   only - corrs   \n",
       "24      H3K9me3                        general   \n",
       "0       H3K27ac                        general   \n",
       "3      H3K27me3                        general   \n",
       "9       H3K4me1                        general   \n",
       "12      H3K4me2                        general   \n",
       "15      H3K4me3                        general   \n",
       "\n",
       "    Modification lncRNAs list count          RBP  RBP lncRNAs count   TP  \\\n",
       "0                              1335    HNRNPLnew                413  172   \n",
       "9                              1290    HNRNPLnew                418  166   \n",
       "0                              1335    HNRNPUnew                559  253   \n",
       "3                               967    HNRNPUnew                562  172   \n",
       "9                              1290    HNRNPUnew                563  221   \n",
       "15                             1168    HNRNPUnew                572  201   \n",
       "17                              268    HNRNPUnew                572   55   \n",
       "0                              1335  HNRNPUL1new                186   91   \n",
       "3                               967  HNRNPUL1new                187   73   \n",
       "6                               879  HNRNPUL1new                187   60   \n",
       "9                              1290  HNRNPUL1new                187   87   \n",
       "15                             1168  HNRNPUL1new                189   77   \n",
       "24                              743  HNRNPUL1new                186   56   \n",
       "0                              1335      SAFBnew                507  238   \n",
       "3                               967      SAFBnew                512  163   \n",
       "9                              1290      SAFBnew                511  215   \n",
       "11                              188      SAFBnew                511   38   \n",
       "15                             1168      SAFBnew                518  191   \n",
       "17                              268      SAFBnew                518   49   \n",
       "24                              743      SAFBnew                509  125   \n",
       "0                              1335     TAF15new                262  128   \n",
       "3                               967     TAF15new                263   95   \n",
       "9                              1290     TAF15new                266  114   \n",
       "12                              823     TAF15new                215   87   \n",
       "15                             1168     TAF15new                267   99   \n",
       "\n",
       "    background    TN    FP   FN       p-value  oddsratio  Correction  \n",
       "0         3769  2193  1163  241  3.209960e-03   1.345768        True  \n",
       "9         3840  2298  1124  252  3.224187e-03   1.346763        True  \n",
       "0         3769  2128  1082  306  1.273295e-07   1.626086        True  \n",
       "3         3837  2480   795  390  9.935819e-04   1.375778        True  \n",
       "9         3840  2208  1069  342  1.337199e-03   1.334712        True  \n",
       "15        3929  2390   967  371  1.440612e-03   1.339040        True  \n",
       "17        3929  3144   213  517  3.740726e-03   1.570273        True  \n",
       "0         3769  2339  1244   95  7.233552e-05   1.801058        True  \n",
       "3         3837  2756   894  114  1.327233e-05   1.974057        True  \n",
       "6         3824  2818   819  127  2.177470e-03   1.625566        True  \n",
       "9         3840  2450  1203  100  1.163485e-04   1.771820        True  \n",
       "15        3929  2649  1091  112  6.114750e-04   1.669283        True  \n",
       "24        3785  2912   687  130  3.013746e-04   1.825910        True  \n",
       "0         3769  2165  1097  269  6.225594e-09   1.746127        True  \n",
       "3         3837  2521   804  349  1.666777e-04   1.464465        True  \n",
       "9         3840  2254  1075  296  1.068543e-05   1.522973        True  \n",
       "11        3840  3179   150  473  4.407285e-03   1.702636        True  \n",
       "15        3929  2434   977  327  1.042747e-04   1.455163        True  \n",
       "17        3929  3192   219  469  8.711866e-03   1.522797        True  \n",
       "24        3785  2658   618  384  1.937878e-03   1.400056        True  \n",
       "0         3769  2300  1207  134  2.536469e-06   1.820228        True  \n",
       "3         3837  2702   872  168  2.974891e-05   1.752198        True  \n",
       "9         3840  2398  1176  152  6.904547e-04   1.529337        True  \n",
       "12        2517  1566   736  128  7.519815e-03   1.446183        True  \n",
       "15        3929  2593  1069  168  4.518960e-03   1.429390        True  "
      ]
     },
     "execution_count": 352,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "union"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.concat([k562, hepG2, union]).to_csv(\"../eCLIP/first_exp_result.tsv\", sep=\"\\t\", index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 349,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_proteins_hep = pd.read_csv(\"../eCLIP/HepG2.Gencode.eCLIP.targets.all.lncRNA.prefer_fixed.tsv\", sep=\"\\t\").columns\n",
    "all_proteins_k562 = pd.read_csv(\"../eCLIP/K562.Gencode.eCLIP.targets.all.lncRNA.prefer_fixed.tsv\", sep=\"\\t\").columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 350,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Gene.ID', 'HNRNPL', 'HNRNPU', 'HNRNPUL1', 'SAFB', 'TAF15'}"
      ]
     },
     "execution_count": 350,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(all_proteins_hep).intersection(set(all_proteins_k562))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 331,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Gene_ID', 'AARS', 'BOP1', 'CPEB4', 'DDX21', 'EXOSC5', 'FASTKD2',\n",
       "       'HNRNPA1', 'HNRNPC', 'HNRNPL', 'HNRNPM', 'HNRNPU', 'HNRNPUL1', 'ILF3',\n",
       "       'KHDRBS1', 'KHSRP', 'LARP4', 'LARP7', 'MATR3', 'NIPBL', 'NPM1', 'NSUN2',\n",
       "       'PPIL4', 'PUS1', 'QKI', 'SAFB2', 'SAFB', 'SBDS', 'SSB', 'SUPV3L1',\n",
       "       'TAF15', 'U2AF1', 'UTP18', 'WDR3', 'WDR43', 'WRN', 'XRN2', 'YWHAG',\n",
       "       'ZC3H8'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 331,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_proteins_k562.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 335,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_proteins_k562 = pd.read_csv(\"../eCLIP/K562.Gencode.eCLIP.targets.all.lncRNA.prefer_fixed.tsv\", sep=\"\\t\")\n",
    "all_proteins_k562 = all_proteins_k562.rename(columns={'Gene.ID': \"Gene_ID\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 336,
   "metadata": {},
   "outputs": [],
   "source": [
    "for p in all_proteins_k562.columns[1:]:\n",
    "    p_df = all_proteins_k562[[\"Gene_ID\", p]]\n",
    "    p_df[p_df[p].notna()].to_csv(\"../eCLIP/\" + p + \"new_K562.txt\", sep=\"\\t\", index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#=================================================================================================================="
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Есть список генов с  сайтами посадки SAFB в K562\n",
    "#Есть список генов с пиками конкретной метки, скоррелированными со списком конкретных РНК\n",
    "#Есть список РНК с которыми вяжется SAFB\n",
    "#1)Аннотировать сайты посадки SAFB - нашей аннотацией, потом все равно пересекать ее\n",
    "#2)Сделать бэкграунд: пересечь все транскрибирующиеся гены в K562(по идее список, которым надо аннотировать сайты) с\n",
    "#нашим списком(которым аннотировали пики)\n",
    "#3)Сделать список lncRNA: пересечь список K562 SAFB с нашим списком lncRNA\n",
    "#4)Есть список генов с сайтом, делаем список генов с пиком конкретной метки, скоррелированный со списком из\n",
    "#предыдущего пункта\n",
    "#)Пересекая эти списки делаем таблицу, потом считаем тест: по тесту на метку"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "def makeCommonGeneSet():\n",
    "    in_file = \"../annotation/gencode.v31.annotation.gff3\"\n",
    "    in_handle = open(in_file)\n",
    "\n",
    "    limit_info = dict(\n",
    "        gff_id = ['chr1', 'chr10', 'chr11', 'chr12', 'chr13', 'chr14', 'chr15', 'chr16', 'chr17', 'chr18', \n",
    "     'chr19', 'chr2', 'chr20', 'chr21', 'chr22', 'chr3', 'chr4', 'chr5', 'chr6', 'chr7', 'chr8', 'chr9', 'chrX', 'chrY'],\n",
    "        gff_type = [\"gene\"])\n",
    "\n",
    "    gencode31_genes = []\n",
    "    for rec in GFF.parse(in_handle, limit_info=limit_info):\n",
    "        gencode31_genes.append(rec.features) \n",
    "\n",
    "    in_handle.close()\n",
    "    \n",
    "    gencode31_genes = [g.qualifiers['gene_id'][0].split('.')[0] for cr in gencode31_genes for g in cr]\n",
    "    \n",
    "    return gencode31_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_res = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "#TODO: А пересекаются ли они в принципе с генами из чипсека?\n",
    "degs = defaultdict(list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TAF15 1129\n",
      "NONO 60\n",
      "HNRNPU 26\n",
      "KHDRBS1 0\n",
      "SAFB2 0\n",
      "HNRNPL 10\n",
      "HNRNPUL1 1\n",
      "HNRNPM 1356\n",
      "SAFB 12\n"
     ]
    }
   ],
   "source": [
    "common_genes = [g.split(\".\")[0] for g in pd.read_csv(\"../eCLIP/\" + \"K562\" + \".eCLIP.detectable.genes.tsv\", header=None)[0].tolist()]\n",
    "deg = pd.read_csv(\"../eCLIP/\" + \"K562\" + \".shRNA.diff.exp.genes.tsv\", sep=\"\\t\")\n",
    "deg[\"Genes\"] = [g.split(\".\")[0] for g in deg[\"Genes\"]]\n",
    "for protein in deg.columns:\n",
    "    if protein == \"Genes\":\n",
    "        continue\n",
    "    print(protein + \" \" + str(len(set([g for g in deg[deg[protein].notna()][\"Genes\"].tolist()]).intersection(common_genes))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MATR3 343\n",
      "NCBP2 285\n",
      "QKI 342\n",
      "TAF15 1934\n",
      "CSTF2T 195\n",
      "FUS 15\n",
      "SLTM 870\n",
      "FAM120A 0\n",
      "CSTF2 37\n",
      "HNRNPA1 136\n",
      "HNRNPC 193\n",
      "HNRNPU 25\n",
      "ILF3 3\n",
      "HNRNPL 20\n",
      "HNRNPUL1 13\n",
      "SFPQ 16\n",
      "NKRF 173\n",
      "SUGP2 188\n",
      "DROSHA 21\n"
     ]
    }
   ],
   "source": [
    "common_genes = [g.split(\".\")[0] for g in pd.read_csv(\"../eCLIP/\" + \"HepG2\" + \".eCLIP.detectable.genes.tsv\", header=None)[0].tolist()]\n",
    "deg = pd.read_csv(\"../eCLIP/\" + \"HepG2\" + \".shRNA.diff.exp.genes.tsv\", sep=\"\\t\")\n",
    "deg[\"Genes\"] = [g.split(\".\")[0] for g in deg[\"Genes\"]]\n",
    "for protein in deg.columns:\n",
    "    if protein == \"Genes\":\n",
    "        continue\n",
    "    print(protein + \" \" + str(len(set([g for g in deg[deg[protein].notna()][\"Genes\"].tolist()]).intersection(common_genes))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TAF15: 41\n",
      "NONO: 18\n",
      "SAFB2: 0\n",
      "HNRNPL: 2\n",
      "HNRNPUL1: 0\n",
      "SAFB: 2\n"
     ]
    }
   ],
   "source": [
    "deg = pd.read_csv(\"../eCLIP/\" + \"K562\" + \".shRNA.diff.exp.genes.tsv\", sep=\"\\t\")\n",
    "deg[\"Genes\"] = [g.split(\".\")[0] for g in deg[\"Genes\"]]\n",
    "for protein in [\"TAF15\", \"NONO\", \"SAFB2\", \"HNRNPL\", \"HNRNPUL1\", \"SAFB\"]:\n",
    "    protein_anno = pd.read_csv(\"../eCLIP/ChIPseq_for_lncRNA_RBPs/\" + \"K562\" + \"/\" + protein + \"_opt_1_not_sorted\" + \"_anno.csv\", sep=\"\\t\")\n",
    "    protein_genes = set([f.split('.')[0] for f in protein_anno['feature']])\n",
    "    protein_deg = [g for g in deg[deg[protein].notna()][\"Genes\"].tolist()]\n",
    "    \n",
    "    print(protein + \": \" + str(len(set(protein_genes).intersection(set(protein_deg)))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SFPQ: 0\n",
      "HNRNPUL1: 1\n"
     ]
    }
   ],
   "source": [
    "deg = pd.read_csv(\"../eCLIP/\" + \"HepG2\" + \".shRNA.diff.exp.genes.tsv\", sep=\"\\t\")\n",
    "deg[\"Genes\"] = [g.split(\".\")[0] for g in deg[\"Genes\"]]\n",
    "common_genes = [g.split(\".\")[0] for g in pd.read_csv(\"../eCLIP/\" + \"HepG2\" + \".eCLIP.detectable.genes.tsv\", header=None)[0].tolist()]\n",
    "for protein in [\"SFPQ\", \"HNRNPUL1\"]:\n",
    "    protein_anno = pd.read_csv(\"../eCLIP/ChIPseq_for_lncRNA_RBPs/\" + \"HepG2\" + \"/\" + protein + \"_opt_1_not_sorted\" + \"_anno.csv\", sep=\"\\t\")\n",
    "    protein_genes = set([f.split('.')[0] for f in protein_anno['feature']]) \n",
    "    protein_deg = [g for g in deg[deg[protein].notna()][\"Genes\"].tolist()]\n",
    "    \n",
    "    print(protein + \": \" + str(len(set(protein_genes).intersection(set(protein_deg)))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Сделать мапку с cell->HM->(protein, anno_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "m = {\"K562\": {\"H3K27ac\": [\"TAF15\", \"HNRNPUL1\", \"SAFB\", \"SAFB2\"], \n",
    "          \"H3K27me3\": [\"TAF15\", \"HNRNPUL1\", \"NONO\"], \n",
    "          \"H3K4me1\": [\"TAF15\", \"HNRNPUL1\", \"SAFB\", \"SAFB2\"], \n",
    "          \"H3K4me3\": [\"HNRNPUL1\", \"SAFB\", \"SAFB2\"], \n",
    "          \"H3K36me3\": [\"HNRNPUL1\"]\n",
    "         },\n",
    "\"HepG2\": {\"H3K27ac\": [\"SFPQ\"],\n",
    "          \"H3K27me3\": [\"SFPQ\"],\n",
    "          \"H3K4me1\":[\"SFPQ\", \"HNRNPUL1\"],\n",
    "          \"H3K4me3\":[\"HNRNPUL1\", \"SFPQ\"],\n",
    "          \"H3K4me2\": [\"HNRNPUL1\"]\n",
    "         }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_res = []\n",
    "all_res_deg = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Get targets list for TAF15new_K562.txt\n",
      "Targets list len 791\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27ac: 104\n",
      "[[510, 87], [17842, 4032]]\n",
      "lalala\n",
      "querying 1-93...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000224959']\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27ac: 104\n",
      "[[1074, 327], [17278, 3792]]\n",
      "lalala\n",
      "querying 1-93...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000224959']\n",
      "Get targets list for HNRNPUL1new_K562.txt\n",
      "Targets list len 855\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27ac: 81\n",
      "[[631, 130], [17501, 4209]]\n",
      "lalala\n",
      "querying 1-73...done.\n",
      "Finished.\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27ac: 81\n",
      "[[1, 0], [18131, 4339]]\n",
      "lalala\n",
      "querying 1-73...done.\n",
      "Finished.\n",
      "Get targets list for SAFBnew_K562.txt\n",
      "Targets list len 1779\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27ac: 144\n",
      "[[4646, 496], [13543, 3786]]\n",
      "lalala\n",
      "querying 1-130...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000279191']\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27ac: 144\n",
      "[[11, 2], [18178, 4280]]\n",
      "lalala\n",
      "querying 1-130...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000279191']\n",
      "Get targets list for SAFB2new_K562.txt\n",
      "Targets list len 2577\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27ac: 241\n",
      "[[58, 3], [19078, 3332]]\n",
      "lalala\n",
      "querying 1-217...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000279191']\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27ac: 241\n",
      "[[0, 0], [19136, 3335]]\n",
      "lalala\n",
      "querying 1-217...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000279191']\n",
      "Get targets list for TAF15new_K562.txt\n",
      "Targets list len 791\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27me3: 74\n",
      "[[285, 312], [10747, 11127]]\n",
      "lalala\n",
      "querying 1-65...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000224959']\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27me3: 74\n",
      "[[647, 754], [10385, 10685]]\n",
      "lalala\n",
      "querying 1-65...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000224959']\n",
      "Get targets list for HNRNPUL1new_K562.txt\n",
      "Targets list len 855\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27me3: 65\n",
      "[[356, 405], [10247, 11463]]\n",
      "lalala\n",
      "querying 1-56...done.\n",
      "Finished.\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27me3: 65\n",
      "[[0, 1], [10603, 11867]]\n",
      "lalala\n",
      "querying 1-56...done.\n",
      "Finished.\n",
      "Get targets list for NONOnew_K562.txt\n",
      "Targets list len 2448\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27me3: 152\n",
      "[[3441, 1885], [9264, 7881]]\n",
      "lalala\n",
      "querying 1-119...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000224959']\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27me3: 152\n",
      "[[41, 29], [12664, 9737]]\n",
      "lalala\n",
      "querying 1-119...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000224959']\n",
      "Get targets list for NONOnew_K562.txt\n",
      "Targets list len 2448\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27me3: 152\n",
      "[[5921, 3133], [6784, 6633]]\n",
      "lalala\n",
      "querying 1-119...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000224959']\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27me3: 152\n",
      "[[41, 29], [12664, 9737]]\n",
      "lalala\n",
      "querying 1-119...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000224959']\n",
      "Get targets list for TAF15new_K562.txt\n",
      "Targets list len 791\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me1: 92\n",
      "[[435, 162], [14962, 6912]]\n",
      "lalala\n",
      "querying 1-77...done.\n",
      "Finished.\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me1: 92\n",
      "[[995, 406], [14402, 6668]]\n",
      "lalala\n",
      "querying 1-77...done.\n",
      "Finished.\n",
      "Get targets list for HNRNPUL1new_K562.txt\n",
      "Targets list len 855\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me1: 74\n",
      "[[594, 167], [15897, 5813]]\n",
      "lalala\n",
      "querying 1-66...done.\n",
      "Finished.\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me1: 74\n",
      "[[1, 0], [16490, 5980]]\n",
      "lalala\n",
      "querying 1-66...done.\n",
      "Finished.\n",
      "Get targets list for SAFBnew_K562.txt\n",
      "Targets list len 1779\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me1: 130\n",
      "[[4382, 760], [12685, 4644]]\n",
      "lalala\n",
      "querying 1-113...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000255455']\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me1: 130\n",
      "[[9, 4], [17058, 5400]]\n",
      "lalala\n",
      "querying 1-113...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000255455']\n",
      "Get targets list for SAFB2new_K562.txt\n",
      "Targets list len 2577\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me1: 232\n",
      "[[40, 21], [18875, 3535]]\n",
      "lalala\n",
      "querying 1-199...done.\n",
      "Finished.\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me1: 232\n",
      "[[0, 0], [18915, 3556]]\n",
      "lalala\n",
      "querying 1-199...done.\n",
      "Finished.\n",
      "Get targets list for HNRNPUL1new_K562.txt\n",
      "Targets list len 855\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me3: 65\n",
      "[[1, 0], [19066, 3404]]\n",
      "lalala\n",
      "querying 1-61...done.\n",
      "Finished.\n",
      "Get targets list for SAFBnew_K562.txt\n",
      "Targets list len 1779\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me3: 107\n",
      "[[5041, 101], [14310, 3019]]\n",
      "lalala\n",
      "querying 1-102...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000279191']\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me3: 107\n",
      "[[12, 1], [19339, 3119]]\n",
      "lalala\n",
      "querying 1-102...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000279191']\n",
      "Get targets list for SAFB2new_K562.txt\n",
      "Targets list len 2577\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me3: 214\n",
      "[[61, 0], [19428, 2982]]\n",
      "lalala\n",
      "querying 1-191...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000279191']\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me3: 214\n",
      "[[0, 0], [19489, 2982]]\n",
      "lalala\n",
      "querying 1-191...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000279191']\n",
      "Get targets list for HNRNPUL1new_K562.txt\n",
      "Targets list len 855\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K36me3: 54\n",
      "[[474, 287], [13139, 8571]]\n",
      "lalala\n",
      "querying 1-43...done.\n",
      "Finished.\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K36me3: 54\n",
      "[[1, 0], [13612, 8858]]\n",
      "lalala\n",
      "querying 1-43...done.\n",
      "Finished.\n",
      "Get targets list for SFPQnew_HepG2.txt\n",
      "Targets list len 1058\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27ac: 146\n",
      "[[310, 30], [19054, 2841]]\n",
      "lalala\n",
      "querying 1-135...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000224959']\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27ac: 146\n",
      "[[16, 1], [19348, 2870]]\n",
      "lalala\n",
      "querying 1-135...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000224959']\n",
      "Get targets list for SFPQnew_HepG2.txt\n",
      "Targets list len 1058\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27me3: 116\n",
      "[[217, 123], [10997, 10898]]\n",
      "lalala\n",
      "querying 1-97...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000224959']\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K27me3: 116\n",
      "[[13, 4], [11201, 11017]]\n",
      "lalala\n",
      "querying 1-97...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000224959']\n",
      "Get targets list for SFPQnew_HepG2.txt\n",
      "Targets list len 1058\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me1: 141\n",
      "[[290, 50], [17788, 4107]]\n",
      "lalala\n",
      "querying 1-126...done.\n",
      "Finished.\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me1: 141\n",
      "[[15, 2], [18063, 4155]]\n",
      "lalala\n",
      "querying 1-126...done.\n",
      "Finished.\n",
      "Get targets list for HNRNPUL1new_HepG2.txt\n",
      "Targets list len 77\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me1: 22\n",
      "[[1716, 946], [9257, 10316]]\n",
      "lalala\n",
      "querying 1-22...done.\n",
      "Finished.\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me1: 22\n",
      "[[5, 8], [10968, 11254]]\n",
      "lalala\n",
      "querying 1-22...done.\n",
      "Finished.\n",
      "Get targets list for HNRNPUL1new_HepG2.txt\n",
      "Targets list len 77\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me3: 20\n",
      "[[2376, 286], [14116, 5457]]\n",
      "lalala\n",
      "querying 1-20...done.\n",
      "Finished.\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me3: 20\n",
      "[[13, 0], [16479, 5743]]\n",
      "lalala\n",
      "querying 1-20...done.\n",
      "Finished.\n",
      "Get targets list for SFPQnew_HepG2.txt\n",
      "Targets list len 1058\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me3: 127\n",
      "[[326, 14], [18908, 2987]]\n",
      "lalala\n",
      "querying 1-121...done.\n",
      "Finished.\n",
      "2 input query terms found no hit:\n",
      "\t['ENSG00000224959', 'ENSG00000272923']\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me3: 127\n",
      "[[15, 2], [19219, 2999]]\n",
      "lalala\n",
      "querying 1-121...done.\n",
      "Finished.\n",
      "2 input query terms found no hit:\n",
      "\t['ENSG00000224959', 'ENSG00000272923']\n",
      "Get targets list for HNRNPUL1new_HepG2.txt\n",
      "Targets list len 77\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me2: 19\n",
      "[[1573, 1089], [7803, 11770]]\n",
      "lalala\n",
      "querying 1-16...done.\n",
      "Finished.\n",
      "trolololo\n",
      "corrs and RBP lncRNAs for H3K4me2: 19\n",
      "[[7, 6], [9369, 12853]]\n",
      "lalala\n",
      "querying 1-16...done.\n",
      "Finished.\n"
     ]
    }
   ],
   "source": [
    "for cell in m.keys():\n",
    "    deg = pd.read_csv(\"../eCLIP/\" + cell + \".shRNA.diff.exp.genes.tsv\", sep=\"\\t\")\n",
    "    deg[\"Genes\"] = [g.split(\".\")[0] for g in deg[\"Genes\"]]\n",
    "    common_genes = [g.split(\".\")[0] for g in pd.read_csv(\"../eCLIP/\" + cell + \".eCLIP.detectable.genes.tsv\", header=None)[0].tolist()]\n",
    "    for target in m[cell].keys():\n",
    "        anno = pd.read_csv(\"/data/mazurovev/all_marks/\" + target + \"/peaks_anno.csv\", sep=\"\\t\") \n",
    "        common_lncRNAs = getCommonGencodeBackground(target, [cell])\n",
    "        for protein in m[cell][target]:\n",
    "            if protein == \"NONO\":\n",
    "                for anno_name in [protein + \"_opt_1_not_sorted\", protein + \"_opt_2_not_sorted\"]:\n",
    "                    protein_anno = pd.read_csv(\"../eCLIP/ChIPseq_for_lncRNA_RBPs/\" + cell + \"/\" + anno_name + \"_anno.csv\", sep=\"\\t\")\n",
    "                    protein_genes = set([f.split('.')[0] for f in protein_anno['feature']]).intersection(common_genes)\n",
    "                    cell_protein_rnas = getTargetsList(protein + \"new_\" + cell + \".txt\")\n",
    "\n",
    "                    res, tp_gene_lists = makeFisher(target, common_genes, common_lncRNAs, cell_protein_rnas, protein_genes)\n",
    "                    df = pd.DataFrame.from_records(res, columns = [\"lncRNA\", \"lncRNA_genes count\", \"protein_genes_count\", \"TP\", \"TN\", 'FP', \"FN\", \"oddsratio\", \"p-value\", \"correction\"])\n",
    "                    df[\"protein\"] = [protein]*df.shape[0]\n",
    "                    df[\"tissue\"] = [cell]*df.shape[0]\n",
    "                    df[\"HM\"] = [target]*df.shape[0]\n",
    "                    df[\"lncRNA_name\"] = get_gene_name(df[\"lncRNA\"])['symbol']\n",
    "                    df = df[df[\"correction\"] == True]\n",
    "                    all_res.append(df)\n",
    "\n",
    "                    if protein in deg.columns:\n",
    "                        deg_res, deg_tp_gene_lists = makeFisher(target, common_genes, common_lncRNAs, cell_protein_rnas, [g for g in deg[deg[protein].notna()][\"Genes\"].tolist()])\n",
    "                        df = pd.DataFrame.from_records(deg_res, columns = [\"lncRNA\", \"lncRNA_genes count\", \"deg_genes_count\", \"TP\", \"TN\", 'FP', \"FN\", \"oddsratio\", \"p-value\", \"correction\"])\n",
    "                        df[\"protein\"] = [protein]*df.shape[0]\n",
    "                        df[\"tissue\"] = [cell]*df.shape[0]\n",
    "                        df[\"HM\"] = [target]*df.shape[0]\n",
    "                        df[\"lncRNA_name\"] = get_gene_name(df[\"lncRNA\"])['symbol']\n",
    "                        df = df[df[\"correction\"] == True]\n",
    "                        all_res_deg.append(df)\n",
    "            else:\n",
    "                anno_name = protein + \"_opt_1_not_sorted\"\n",
    "                protein_anno = pd.read_csv(\"../eCLIP/ChIPseq_for_lncRNA_RBPs/\" + cell + \"/\" + anno_name + \"_anno.csv\", sep=\"\\t\")\n",
    "                protein_genes = set([f.split('.')[0] for f in protein_anno['feature']]).intersection(common_genes)\n",
    "                cell_protein_rnas = getTargetsList(protein + \"new_\" + cell + \".txt\")\n",
    "\n",
    "                res, tp_gene_lists = makeFisher(target, common_genes, common_lncRNAs, cell_protein_rnas, protein_genes)\n",
    "                df = pd.DataFrame.from_records(res, columns = [\"lncRNA\", \"lncRNA_genes count\", \"protein_genes_count\", \"TP\", \"TN\", 'FP', \"FN\", \"oddsratio\", \"p-value\", \"correction\"])\n",
    "                df[\"protein\"] = [protein]*df.shape[0]\n",
    "                df[\"tissue\"] = [cell]*df.shape[0]\n",
    "                df[\"HM\"] = [target]*df.shape[0]\n",
    "                df[\"lncRNA_name\"] = get_gene_name(df[\"lncRNA\"])['symbol']\n",
    "                df = df[df[\"correction\"] == True]\n",
    "                all_res.append(df)\n",
    "\n",
    "                if protein in deg.columns:\n",
    "                    deg_res, deg_tp_gene_lists = makeFisher(target, common_genes, common_lncRNAs, cell_protein_rnas, [g for g in deg[deg[protein].notna()][\"Genes\"].tolist()])\n",
    "                    df = pd.DataFrame.from_records(deg_res, columns = [\"lncRNA\", \"lncRNA_genes count\", \"deg_genes_count\", \"TP\", \"TN\", 'FP', \"FN\", \"oddsratio\", \"p-value\", \"correction\"])\n",
    "                    df[\"protein\"] = [protein]*df.shape[0]\n",
    "                    df[\"tissue\"] = [cell]*df.shape[0]\n",
    "                    df[\"HM\"] = [target]*df.shape[0]\n",
    "                    df[\"lncRNA_name\"] = get_gene_name(df[\"lncRNA\"])['symbol']\n",
    "                    df = df[df[\"correction\"] == True]\n",
    "                    all_res_deg.append(df)\n",
    "                                     \n",
    "                                     "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "fantom = set(['A1BG-AS1',\n",
    " 'AC005592.2',\n",
    " 'AC007246.3',\n",
    " 'AC007879.7',\n",
    " 'AC016747.3',\n",
    " 'AC092295.7',\n",
    " 'BOLA3-AS1',\n",
    " 'CD27-AS1',\n",
    " 'CTD-2587H24.5',\n",
    " 'DNM3OS',\n",
    " 'EMX2OS',\n",
    " 'FGD5-AS1',\n",
    " 'FTX',\n",
    " 'JPX',\n",
    " 'LINC00511',\n",
    " 'LINC00654',\n",
    " 'LINC00702',\n",
    " 'LINC00707',\n",
    " 'LINC00862',\n",
    " 'LINC00886',\n",
    " 'LINC00938',\n",
    " 'LINC00963',\n",
    " 'MAPKAPK5-AS1',\n",
    " 'MEG3',\n",
    " 'RAB30-AS1',\n",
    " 'RP11-115C21.2',\n",
    " 'RP11-137L10.6',\n",
    " 'RP11-150O12.1',\n",
    " 'RP11-221N13.3',\n",
    " 'RP11-38L15.3',\n",
    " 'RP11-395B7.4',\n",
    " 'RP11-398K22.12',\n",
    " 'RP11-417E7.1',\n",
    " 'RP11-458D21.1',\n",
    " 'RP11-539L10.3',\n",
    " 'RP11-545E17.3',\n",
    " 'RP11-54A9.1',\n",
    " 'RP11-65J3.1',\n",
    " 'RP11-660L16.2',\n",
    " 'RP11-834C11.4',\n",
    " 'RP13-463N16.6',\n",
    " 'RP3-510D11.2',\n",
    " 'RP6-109B7.3',\n",
    " 'SBF2-AS1',\n",
    " 'SERTAD4-AS1',\n",
    " 'TMCC1-AS1',\n",
    " 'WDFY3-AS2',\n",
    " 'ZNF674-AS1'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "eclip = set(pd.concat(all_res)['lncRNA_name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'LINC00511', 'MEG3'}"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fantom.intersection(eclip)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "48"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(fantom)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "deg = pd.read_csv(\"../eCLIP/K562.shRNA.diff.exp.genes.tsv\", sep=\"\\t\")\n",
    "deg[\"Genes\"] = [g.split(\".\")[0] for g in deg[\"Genes\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genes</th>\n",
       "      <th>TAF15</th>\n",
       "      <th>NONO</th>\n",
       "      <th>HNRNPU</th>\n",
       "      <th>KHDRBS1</th>\n",
       "      <th>SAFB2</th>\n",
       "      <th>HNRNPL</th>\n",
       "      <th>HNRNPUL1</th>\n",
       "      <th>HNRNPM</th>\n",
       "      <th>SAFB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000971</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000001036</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Genes  TAF15  NONO  HNRNPU  KHDRBS1  SAFB2  HNRNPL  HNRNPUL1  \\\n",
       "0  ENSG00000000419    NaN   NaN     NaN      NaN    NaN     NaN       NaN   \n",
       "1  ENSG00000000457    NaN   NaN     NaN      NaN    NaN     NaN       NaN   \n",
       "2  ENSG00000000460    NaN   NaN     NaN      NaN    NaN     NaN       NaN   \n",
       "3  ENSG00000000971    NaN   NaN     NaN      NaN    NaN     NaN       NaN   \n",
       "4  ENSG00000001036    NaN   NaN     NaN      NaN    NaN     NaN       NaN   \n",
       "\n",
       "   HNRNPM  SAFB  \n",
       "0     NaN   NaN  \n",
       "1     NaN   NaN  \n",
       "2     NaN   NaN  \n",
       "3     NaN   NaN  \n",
       "4     NaN   NaN  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "deg.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "safb_deg = [g for g in deg[deg[\"SAFB\"].notna()][\"Genes\"].tolist()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ENSG00000104228',\n",
       " 'ENSG00000104679',\n",
       " 'ENSG00000106246',\n",
       " 'ENSG00000116521',\n",
       " 'ENSG00000118515',\n",
       " 'ENSG00000130513',\n",
       " 'ENSG00000143436',\n",
       " 'ENSG00000151239',\n",
       " 'ENSG00000160679',\n",
       " 'ENSG00000163113',\n",
       " 'ENSG00000163704',\n",
       " 'ENSG00000165813',\n",
       " 'ENSG00000225830']"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "safb_deg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "for rna in df[\"lncRNA\"]:\n",
    "    print(len(tp_gene_lists[rna].intersection(safb_deg)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.concat(all_res_deg).to_csv(\"../eCLIP/result_with_ChIP_and_degs_09.11.2020.tsv\", sep=\"\\t\", index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 408,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mazurovev/.local/lib/python3.6/site-packages/ipykernel_launcher.py:1: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    }
   ],
   "source": [
    "pd.concat(all_res).to_csv(\"../eCLIP/result_with_ChIP_26.10.2020.tsv\", sep=\"\\t\", index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Брать в качестве генов только те, в которых есть пики, сколько их?\n",
    "#anno = pd.read_csv(\"/data/mazurovev/all_marks/\" + \"H3K27me3\" + \"/peaks_anno.csv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 256,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Общие гены все(не только нкРНК)\n",
    "# common_genes = [g.split(\".\")[0] for g in anno['feature'].unique()]\n",
    "\n",
    "#В качестве общих генов берем экспрессирующиеся\n",
    "#common_genes = [g.split(\".\")[0] for g in pd.read_csv(\"../eCLIP/\" + \"K562\" + \".eCLIP.detectable.genes.tsv\", header=None)[0].tolist()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "metadata": {},
   "outputs": [],
   "source": [
    "#lncRNAs background\n",
    "#common_lncRNAs = getCommonGencodeBackground(\"H3K27ac\", [\"K562\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Аннотация чипсека safb для K562\n",
    "#safb_anno = pd.read_csv(\"../eCLIP/SAFB_anno.csv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Список генов, которые входят в общий с нами список генов и в чипсик белка\n",
    "#safb_genes = set([f.split('.')[0] for f in safb_anno['feature']]).intersection(common_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 264,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Get targets list for SAFBnew_K562.txt\n",
      "Targets list len 1779\n"
     ]
    }
   ],
   "source": [
    "#нкРНК мишени у SAFB в K562(и которые могут быть в корреляциях)\n",
    "#k562_SAFB_rnas = getTargetsList(\"SAFBnew_K562.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "def makeFisher(target, common_genes, common_lncRNAs, protein_cell_rnas, protein_genes):\n",
    "    tp_gene_lists = {}\n",
    "    print(\"trolololo\")\n",
    "    #нкРНК с корреляциями у H3K27ac(и которые могут быть в K562)\n",
    "    target_rnas = set(corrs_lncRNAs[target]['general']).intersection(common_lncRNAs)\n",
    "    \n",
    "    #нкРНК, которые и имеют корреляции с меткой и есть в мишенях белка\n",
    "    lncRNAs = protein_cell_rnas.intersection(target_rnas)\n",
    "    print(\"corrs and RBP lncRNAs for \" + target + \": \" + str(len(lncRNAs)))\n",
    "    \n",
    "    pg_association = pd.read_csv(\"/data/mazurovev/all_marks/\" + target + \"/lncRNA_peaks_gene_association.tsv\", sep=\"\\t\")\n",
    "    res_per_lncRNA = []\n",
    "    all_lncRNAs_hm_genes = set([g.split('.')[0] for g in pg_association[pg_association['lncRNA'].isin(lncRNAs)]['gene'].unique()]).intersection(common_genes)\n",
    "    for lncRNA in lncRNAs:\n",
    "        #Список генов(общий список с генами белка), которые имеют корреляцию с меткой через нкРНК, которые так же мишени белка\n",
    "        hm_genes = set([g.split('.')[0] for g in pg_association[pg_association['lncRNA'] == lncRNA]['gene'].unique()]).intersection(common_genes)\n",
    "        if len(hm_genes) > 0:\n",
    "            TP = len(hm_genes.intersection(protein_genes))\n",
    "            tp_gene_lists[lncRNA] = hm_genes.intersection(protein_genes)\n",
    "            TN = len(common_genes) - len(protein_genes) - len(hm_genes) + TP\n",
    "            FP = len(protein_genes) - TP\n",
    "            FN = len(hm_genes) - TP\n",
    "            r, pv = stats.fisher_exact([[TP, FP], [FN, TN]], alternative='greater')\n",
    "            res_per_lncRNA.append([lncRNA, len(hm_genes), len(protein_genes), TP, TN, FP, FN, r, pv])\n",
    "    \n",
    "    TP = len(all_lncRNAs_hm_genes.intersection(protein_genes))\n",
    "    TN = len(common_genes) - len(protein_genes) - len(all_lncRNAs_hm_genes) + TP\n",
    "    FP = len(protein_genes) - TP\n",
    "    FN = len(all_lncRNAs_hm_genes) - TP\n",
    "    print([[TP, FP], [FN, TN]])\n",
    "    print(\"lalala\")\n",
    "    r, pv = stats.fisher_exact([[TP, FP], [FN, TN]], alternative='greater')\n",
    "    # res_per_lncRNA.append([\"All\", len(all_lncRNAs_hm_genes), len(protein_genes), TP, TN, FP, FN, r, pv])\n",
    "    \n",
    "    pv_list = [r[8] for r in res_per_lncRNA]\n",
    "    correct = multipletests(pv_list, alpha=0.05, method='fdr_bh')\n",
    "    for i, c in enumerate(correct[0]):\n",
    "        res_per_lncRNA[i].append(c)\n",
    "    \n",
    "    return res_per_lncRNA, tp_gene_lists"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 293,
   "metadata": {},
   "outputs": [],
   "source": [
    "safb_deg = deg[[\"Genes\", \"SAFB\"]]\n",
    "safb_deg = [g.split(\".\")[0] for g in safb_deg[safb_deg[\"SAFB\"].notna()][\"Genes\"].tolist()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 294,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13"
      ]
     },
     "execution_count": 294,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(safb_deg)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 296,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ENSG00000116521'}"
      ]
     },
     "execution_count": 296,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(tp_gene_lists['ENSG00000197536']).intersection(set(safb_deg))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "import mygene\n",
    "def get_gene_name(gene_id):\n",
    "    mg = mygene.MyGeneInfo()\n",
    "    out = mg.querymany(gene_id, scopes='ensembl.gene', fields=\"symbol\", species='human', \n",
    "                           returnall=True, as_dataframe=True)\n",
    "    with_symbol = out['out']\n",
    "    del with_symbol.index.name\n",
    "    with_symbol['ensembl_id'] = with_symbol.index\n",
    "    if 'symbol' in with_symbol.columns:\n",
    "        with_symbol = with_symbol.reset_index()[['ensembl_id', 'symbol']]\n",
    "        with_symbol = with_symbol[with_symbol['symbol'].notnull()]\n",
    "        return with_symbol\n",
    "    else:\n",
    "        return None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
