{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import rpy2.robjects as robjects\n",
    "import pandas as pd\n",
    "from multiprocessing import Pool\n",
    "import pickle\n",
    "import os\n",
    "from subprocess import Popen, PIPE\n",
    "import shutil\n",
    "from functools import partial\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <span>ListVector with 2 elements.</span>\n",
       "    <table>\n",
       "      <tbody>\n",
       "      \n",
       "      <tr>\n",
       "      <th>\n",
       "        value\n",
       "      </th>\n",
       "      <td>\n",
       "        function(path, file_prefix) {\n",
       "  print(paste(\"find files in \", path, \" with prefix: \", file_prefix, sep=\"\"))\n",
       "  files <- list.files(path=path, file_prefix)\n",
       "  print(files)\n",
       "  res <- list()\n",
       "  for(f in files) {\n",
       "    print(paste(\"read file \", path, f, sep=\"\"))\n",
       "    res <- append(res, NCIS:::read.BED(paste(path, f, sep=\"\")))\n",
       "    print(paste(\"append \", f, sep = \"\"))\n",
       "  }\n",
       "  \n",
       "  return(res)\n",
       "}\n",
       "\n",
       "      </td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "      <th>\n",
       "        visible\n",
       "      </th>\n",
       "      <td>\n",
       "        \n",
       "    <span>BoolVector with 1 elements.</span>\n",
       "    <table>\n",
       "      <tbody>\n",
       "      <tr>\n",
       "      \n",
       "      <td>\n",
       "               0\n",
       "      </td>\n",
       "      \n",
       "      </tr>\n",
       "      </tbody>\n",
       "    </table>\n",
       "    \n",
       "      </td>\n",
       "      </tr>\n",
       "      \n",
       "      </tbody>\n",
       "    </table>\n",
       "    "
      ],
      "text/plain": [
       "R object with classes: ('list',) mapped to:\n",
       "<ListVector - Python:0x7f38f8354ac8 / R:0x5b25e98>\n",
       "[SignatureTranslatedFunc..., BoolVector]\n",
       "  value: <class 'rpy2.robjects.functions.SignatureTranslatedFunction'>\n",
       "  R object with classes: ('function',) mapped to:\n",
       "<SignatureTranslatedFunction - Python:0x7f38f8355cc8 / R:0x5b51d58>\n",
       "  visible: <class 'rpy2.robjects.vectors.BoolVector'>\n",
       "  R object with classes: ('logical',) mapped to:\n",
       "<BoolVector - Python:0x7f38f8354688 / R:0x2acf738>\n",
       "[       0]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r = robjects.r\n",
    "r.source(\"bin_data_for_diagnostics_plot.R\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "def createChroms(path, signal, control, signal_path, control_path):\n",
    "    #0. Создаем папку\n",
    "    if not os.path.exists(path):\n",
    "        os.makedirs(path)\n",
    "        os.chmod(path, 0o777)\n",
    "    #Все bed файлы должны быть отсортированны\n",
    "    print(signal)\n",
    "    try:\n",
    "        callWithNiceOutput(\"for chr in `bedextract --list-chr \" + signal_path + signal + \"`; do bedextract $chr \" + signal_path + signal + \"> \" + path + signal + \".$chr; done\")\n",
    "        callWithNiceOutput(\"for chr in `bedextract --list-chr \" + control_path + control + \"`; do bedextract $chr \" + control_path + control + \"> \" + path + control + \".$chr; done\")\n",
    "        #subprocess.check_output(\"for chr in `bedextract --list-chr \" + signal_path + signal + \"`; do bedextract $chr \" + signal_path + signal + \"> \" + path + signal + \".$chr; done\", shell=True)\n",
    "        #subprocess.check_output(\"for chr in `bedextract --list-chr \" + control_path + control + \"`; do bedextract $chr \" + control_path + control + \"> \" + path + control + \".$chr; done\", shell=True)\n",
    "        \n",
    "        print(\"remove ugly chroms\")\n",
    "        files = [f for f in os.listdir(path) if not re.match(r'(' + signal + '|' + control +')\\.chr[M|X|Y|[0-9][0-9]?$', f)]\n",
    "        for f in files:\n",
    "            os.remove(path + f) \n",
    "    except subprocess.CalledProcessError as e:\n",
    "        raise RuntimeError(\"command '{}' return with error (code {}): {}\".format(e.cmd, e.returncode, e.output))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def callWithNiceOutput(cmd):\n",
    "    proc = Popen(cmd, stdout=PIPE, stderr=PIPE, shell=True, bufsize=-1)\n",
    "    \n",
    "    for line in proc.stdout:\n",
    "        print(\"output: \")\n",
    "        print(line)\n",
    "        sys.stdout.flush()\n",
    "    \n",
    "    for line in proc.stderr:\n",
    "        print(\"error: \")\n",
    "        print(line)\n",
    "        sys.stdout.flush()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "#est : the estimated normalizing factor.\n",
    "\n",
    "#binsize.est : the estimated binsize.\n",
    "\n",
    "#r.seq.depth: sequencing depth ratio.\n",
    "\n",
    "#pi0: the estimated proportion of background reads among ChIP sample.\n",
    "\n",
    "def makeNormalizationFactorAndPlot(signal_path, control_path, signal_name, control_name):\n",
    "    signal = signal_name + \".bed\"\n",
    "    control = control_name + \".bed\"\n",
    "    path = \"/data/mazurovev/norm_factors/f_\" + signal_name + \"/\"\n",
    "    \n",
    "    #createChroms(path, signal, control, signal_path, control_path)\n",
    "    print(\"start count factor for : \" + signal)\n",
    "    res = r.countFactorAndDiagnosticPlot(path, path, signal + \".chr\", control + \".chr\",\n",
    "                                         \"/data/mazurovev/norm_factors/plots/\" + signal_name)\n",
    "    #python_res = dict(zip(res.names, map(list,list(res))))\n",
    "    #python_res['chip'] = data[2]\n",
    "    \n",
    "    #with open('/data/mazurovev/norm_factors/plots/' + signal_name + \".txt\", 'w') as f:\n",
    "    #    print(python_res, file=f)\n",
    "        \n",
    "    print(\"done with \" + signal_name)\n",
    "    #shutil.rmtree(path)\n",
    "    #print(\"delete \" + path)\n",
    "    #return python_res"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "def run(data_path, res_path):\n",
    "    d = pd.read_csv(data_path, sep='\\t')\n",
    "\n",
    "    data = []\n",
    "    exist = [f.split('.')[0] for f in os.listdir(\"/data/mazurovev/norm_factors/plots\") if f.split('.')[-1] == 'txt']\n",
    "    for index, row in d.iterrows():\n",
    "        target = row['Experiment target'].split('-')[0]\n",
    "        file_path = '/data/mazurovev/' + target + \"/\" + row['Biosample term id'] + \"/\"\n",
    "        control_path = '/data/mazurovev/controls/'\n",
    "        if(row['File accession'] not in exist):\n",
    "            data.append((file_path, control_path, row['File accession'], row['Control']))\n",
    " \n",
    "    pool = Pool(processes=25)\n",
    "    factors = pool.starmap(makeNormalizationFactorAndPlot, data)\n",
    "    #print(factors)\n",
    "    pool.close()\n",
    "    pool.join()\n",
    "    \n",
    "    #with open(res_path, 'wb') as f:\n",
    "    #    pickle.dump(factors, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def addNormFactorsToFile(pickle_path, csv_path):\n",
    "    d = pd.read_csv(csv_path, sep=';', index_col=False)\n",
    "    with open(pickle_path, 'rb') as f:\n",
    "        data = pickle.load(f)\n",
    "        \n",
    "    factors_col = []\n",
    "    for index, row in d.iterrows():\n",
    "        factors_col.append([i['est'] for i in data if i['chip'] == row['File accession']][0][0])\n",
    "        \n",
    "    d['norm_factor'] = factors_col\n",
    "    d.to_csv(csv_path, sep=';', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Считаем факторы для H3K27me3\n",
    "#run(\"../H3K27me3/H3K27me3_filtered_by_biosample.csv\", \"../H3K27me3/normalizationFactors.pickle\", \"../H3K27me3/bw/\", \"../H3K27me3/control/\")\n",
    "#addNormFactorsToFile(\"../H3K27me3/normalizationFactors.pickle\", \"../H3K27me3/H3K27me3_filtered_by_biosample.csv\")\n",
    "\n",
    "#Считаем факторы для H3K4me1\n",
    "run(\"../ChiP-Seq_for_all_marks_all_replicas.tsv\", \"/data/mazurovEV/norm_factors/normalizationFactors.pickle\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = pd.read_csv(\"../ChiP-Seq_for_all_marks_all_replicas.tsv\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENCFF186TYR without norm factor!\n",
      "ENCFF843MXE without norm factor!\n"
     ]
    }
   ],
   "source": [
    "est = []\n",
    "binsize = []\n",
    "seq_depth = []\n",
    "pi0 = []\n",
    "for i, row in d.iterrows():\n",
    "    file_name = \"/data/mazurovev/norm_factors/plots/\" + row['File accession'] + \".bed.chr.txt\"\n",
    "    if(os.path.isfile(file_name)):\n",
    "        fh = open(file_name, \"r\") \n",
    "        strr = fh.read().split(' ')\n",
    "        est.append(float(strr[0]))\n",
    "        binsize.append(float(strr[1]))\n",
    "        seq_depth.append(float(strr[2]))\n",
    "        pi0.append(float(strr[3]))\n",
    "    else:\n",
    "        est.append(-1)\n",
    "        binsize.append(-1)\n",
    "        seq_depth.append(-1)\n",
    "        pi0.append(-1)\n",
    "        print(row['File accession'] + \" without norm factor!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "d['norm_factor'] = est\n",
    "d['binsize'] = binsize\n",
    "d['seq_depth'] = seq_depth\n",
    "d['pi0'] = pi0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "d.to_csv(\"../ChiP-Seq_for_all_marks_all_replicas.tsv\", sep=\"\\t\", index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>File accession</th>\n",
       "      <th>File format</th>\n",
       "      <th>Output type</th>\n",
       "      <th>Experiment accession</th>\n",
       "      <th>Biosample term id</th>\n",
       "      <th>Experiment target</th>\n",
       "      <th>Biological replicate(s)</th>\n",
       "      <th>Technical replicate</th>\n",
       "      <th>Read length</th>\n",
       "      <th>Mapped read length</th>\n",
       "      <th>...</th>\n",
       "      <th>Paired with</th>\n",
       "      <th>File download URL</th>\n",
       "      <th>Assembly</th>\n",
       "      <th>File Status</th>\n",
       "      <th>Lab</th>\n",
       "      <th>Control</th>\n",
       "      <th>norm_factor</th>\n",
       "      <th>binsize</th>\n",
       "      <th>seq_depth</th>\n",
       "      <th>pi0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>ENCFF186TYR</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR037SNV</td>\n",
       "      <td>UBERON:0000992</td>\n",
       "      <td>H3K27me3-human</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF186TY...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>released</td>\n",
       "      <td>ENCODE Processing Pipeline</td>\n",
       "      <td>ENCFF791PRR</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>895</th>\n",
       "      <td>ENCFF843MXE</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR364YAY</td>\n",
       "      <td>UBERON:0004550</td>\n",
       "      <td>H3K36me3-human</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>76.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF843MX...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>released</td>\n",
       "      <td>ENCODE Processing Pipeline</td>\n",
       "      <td>ENCFF584UYW</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    File accession File format Output type Experiment accession  \\\n",
       "32     ENCFF186TYR         bam  alignments          ENCSR037SNV   \n",
       "895    ENCFF843MXE         bam  alignments          ENCSR364YAY   \n",
       "\n",
       "    Biosample term id Experiment target  Biological replicate(s)  \\\n",
       "32     UBERON:0000992    H3K27me3-human                        1   \n",
       "895    UBERON:0004550    H3K36me3-human                        1   \n",
       "\n",
       "     Technical replicate  Read length  Mapped read length ...   Paired with  \\\n",
       "32                   NaN          NaN                36.0 ...           NaN   \n",
       "895                  NaN          NaN                76.0 ...           NaN   \n",
       "\n",
       "                                     File download URL Assembly File Status  \\\n",
       "32   https://www.encodeproject.org/files/ENCFF186TY...   GRCh38    released   \n",
       "895  https://www.encodeproject.org/files/ENCFF843MX...   GRCh38    released   \n",
       "\n",
       "                            Lab      Control norm_factor  binsize  seq_depth  \\\n",
       "32   ENCODE Processing Pipeline  ENCFF791PRR        -1.0     -1.0       -1.0   \n",
       "895  ENCODE Processing Pipeline  ENCFF584UYW        -1.0     -1.0       -1.0   \n",
       "\n",
       "     pi0  \n",
       "32  -1.0  \n",
       "895 -1.0  \n",
       "\n",
       "[2 rows x 21 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d[d['File accession'].isin(['ENCFF186TYR', 'ENCFF843MXE'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
